---
title: "acute_vs_chronic"
author: "Becca Belmonte"
date: "2022-12-27"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Libraries
```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(DESeq2)
library(ggrepel)
library(WGCNA)
library(viridis)
library(clusterProfiler)
library(enrichplot)
library(ggplot2)
library(plotly)
organism = "org.Dm.eg.db"
#BiocManager::install(organism, character.only = TRUE)
library(organism, character.only = TRUE)
select <- dplyr::select
```


```{r Colors, number=TRUE}
#viridis(10,option="magma")
colors_time <- c("#A02F7F","#F56B5C", "#FEC287", "#000004")
colors_sex <- c("#D3C164", "#05366E")
colors_inf <- c("#000004", "#FEC98D")
colors_wound <- c("#F1605D", "#FEC98D")
colors_dis <- c("#451077", "#9F2F7F")
colors_hem <- viridis(4, option = "cividis")

text_sex <- c("Female", "Male")
text_inf <- c("Infected", "Uninfected")
text_wound <- c("Wounded", "Unchallenged")
text_dis <- c("Dissected", "Not Dissected")
text_hem <- c("Internalization", "Maturation", "Migration", "Receptors")
```


```{r Raw Data, number=TRUE}
countdata <- read.csv("Data/david_counts_wide_unambiguous.csv", row.names = 1, sep = ";")

countdata[is.na(countdata)] <- 0

remove <- c("XF:Z:alignment_not_unique", "XF:Z:no_feature")
rrna <- rownames(countdata)[grep("rRNA", rownames(countdata))]

countdata <- countdata[!rownames(countdata) %in% remove,]
countdata <- countdata[!rownames(countdata) %in% rrna,]

t_countdata <- as.data.frame(t(countdata))
t_countdata$ID <- rownames(t_countdata)

treatmentdata <- read.csv("Data/RNAseq_SexDim_ID.csv", sep=";", row.names = 1)
treatmentdata$ID <- rownames(treatmentdata)
not_dissected <- c("RG.Z.sample_165", "RG.Z.sample_61", "RG.Z.sample_166")
dissected <- c("RG.Z.sample_147")

# based on spermatogenesis gene expression (fzo & Dpy-30L2), sample 165 is unambiguously not dissected -- I correct.
treatmentdata$Dissected_Y.N = ifelse(treatmentdata$ID %in% not_dissected,"N",as.character(treatmentdata$Dissected_Y.N))
treatmentdata$Dissected_Y.N = ifelse(treatmentdata$ID %in% dissected,"Y",as.character(treatmentdata$Dissected_Y.N))

#nrow(subset(treatmentdata, Dissected_Y.N=="N"))

#treatmentdata %>% 
#  group_by(Dissected_Y.N) %>% 
#  summarise(rows = n()) 
#treatmentdata %>% 
#  group_by(Treatment) %>% 
#  summarise(rows = n())
#treatmentdata %>% 
#  group_by(Sex) %>% 
#  summarise(rows = n())
#treatmentdata %>% 
#  group_by(Rep) %>% 
 # summarise(rows = n())

RNAseq_data <- left_join(treatmentdata,t_countdata)

RNAseq_data$totalcounts <- rowSums(RNAseq_data[,-(1:8)])

RNAseq_TPM_col <- RNAseq_data[,(1:8)]

RNAseq_TPM_data <- RNAseq_data[,9:16159]/RNAseq_data$totalcounts*10^6
#101.92322
RNAseq_TPM_data <- cbind(RNAseq_TPM_col, RNAseq_TPM_data)

matrix_countdata <- as.matrix(countdata)
colData <- treatmentdata %>% 
  arrange(ID)
rownames(colData) <- colData$ID
#colnames(matrix_countdata) %in% rownames(colData)
colData <- colData %>% 
  dplyr::select(Rep:Dissected_Y.N, ID)

colData <- colData %>% 
  mutate(Group = str_c(Dissected_Y.N, Sex, Treatment, time, Rep, sep = "_"))
rownames(colData) <- colData$ID
colnames(matrix_countdata) <- colData$Group
rownames(colData) <- colData$Group
colData$Rep <- as.factor(colData$Rep)
#write.csv(colData, file = "colData.csv")

countdata <- countdata %>% 
  select(colData$ID)
colnames(countdata) <- colData$Group
rownames(countdata) <- sub("XF:Z:", "", rownames(countdata))


Gene_ID_Flybase <- read.table("Data/fbgn_annotation_ID_fb_2020_01.tsv",header=F,sep="\t")
colnames(Gene_ID_Flybase) <- c("Gene_name","Species","FBgn_ID","Former_FBgn_ID","CG_ID","Former_CG_ID")
Gene_ID_Flybase <- Gene_ID_Flybase %>% 
  filter(Species == "Dmel")
Gene_ID_Flybase$Species <- as.factor(Gene_ID_Flybase$Species)

Drosophila_KEGG <- read.delim("Data/Drosophila_KEGG.gmt")

t_Drosophila_KEGG <- as.data.frame(t(Drosophila_KEGG))
colnames(t_Drosophila_KEGG) <- t_Drosophila_KEGG[1,]
t_Drosophila_KEGG <- as.list(t_Drosophila_KEGG[-c(1,2), ])


Drosophila_REACTOME <- read.delim("Data/Drosophila_REACTOME.gmt")

t_Drosophila_REACTOME <- as.data.frame(t(Drosophila_REACTOME))
colnames(t_Drosophila_REACTOME) <- t_Drosophila_REACTOME[1,]
t_Drosophila_REACTOME <- as.list(t_Drosophila_REACTOME[-c(1,2), ])
```

# DE models
```{r}
############################################################
# Setting up the dissected 0, 8, 24, 72 model

colData$Dissected_Y.N <- as.factor(colData$Dissected_Y.N)
colData$Dissected_Y.N <- relevel(factor(colData$Dissected_Y.N), "Y")
colData_pr <- colData %>% 
  filter(!Treatment == "WD" & Dissected_Y.N == "Y") %>% 
  mutate(Treatment = factor(Treatment, levels = c("UC", "Pr")),
         Sex = factor(Sex, levels = c("F", "M")))

times <- c("8", "24", "72")
top_each_time <- vector("list")
for(i in times){
  colData_loop <- colData_pr %>% 
  filter(time == i)
countdata_loop <- countdata %>% 
  select(colData_loop$Group)
matrix_countdata_loop <- as.matrix(countdata_loop)

dds_loop <- DESeqDataSetFromMatrix(countData = matrix_countdata_loop,
colData = colData_loop,
design = ~  Sex + Treatment)



dds_loop <- estimateSizeFactors(dds_loop)

dds_loop <- DESeq(dds_loop, test="LRT", reduced = ~ Sex)
resTC_loop <- DESeq2::results(dds_loop)

tab_resTC_loop <- as.data.frame(resTC_loop)
tab_resTC_loop$gene_name <- rownames(tab_resTC_loop) 
tab_resTC_loop$gene_name <- sub("XF:Z:","",tab_resTC_loop$gene_name)


tab_resTC_loop <- cbind(tab_resTC_loop,Gene_ID_Flybase[match(tab_resTC_loop$gene_name,Gene_ID_Flybase$Gene_name),"FBgn_ID"])
colnames(tab_resTC_loop)[ncol(tab_resTC_loop)] ="FBgn_ID"

top_each_time[[i]] <- tab_resTC_loop
}
top_8 <- top_each_time[["8"]]
top_24 <- top_each_time[["24"]]
top_72 <- top_each_time[["72"]]




###########################################
# Sex specific responses over time

times <- c("8", "24", "72")
top_each_time_inter <- vector("list")
for(i in times){
  colData_loop <- colData_pr %>% 
  filter(time == i)
countdata_loop <- countdata %>% 
  select(colData_loop$Group)
matrix_countdata_loop <- as.matrix(countdata_loop)

dds_loop <- DESeqDataSetFromMatrix(countData = matrix_countdata_loop,
colData = colData_loop,
design = ~  Sex*Treatment)



dds_loop <- estimateSizeFactors(dds_loop)

dds_loop <- DESeq(dds_loop, test="LRT", reduced = ~ Sex + Treatment)
resTC_loop <- DESeq2::results(dds_loop)

tab_resTC_loop <- as.data.frame(resTC_loop)
tab_resTC_loop$gene_name <- rownames(tab_resTC_loop) 
tab_resTC_loop$gene_name <- sub("XF:Z:","",tab_resTC_loop$gene_name)


tab_resTC_loop <- cbind(tab_resTC_loop,Gene_ID_Flybase[match(tab_resTC_loop$gene_name,Gene_ID_Flybase$Gene_name),"FBgn_ID"])
colnames(tab_resTC_loop)[ncol(tab_resTC_loop)] ="FBgn_ID"

top_each_time_inter[[i]] <- tab_resTC_loop
}
top_inter_8 <- top_each_time_inter[["8"]]
top_inter_24 <- top_each_time_inter[["24"]]
top_inter_72 <- top_each_time_inter[["72"]]


# Loop response at each time point (females)
top_each_time_F <- vector("list")

times <- c("8", "24", "72")
for(i in times){
  colData_loop <- colData_pr %>% 
  filter(time == i & Sex == "F")
countdata_loop <- countdata %>% 
  select(colData_loop$Group)
matrix_countdata_loop <- as.matrix(countdata_loop)

dds_loop <- DESeqDataSetFromMatrix(countData = matrix_countdata_loop,
colData = colData_loop,
design = ~  Treatment)



dds_loop <- estimateSizeFactors(dds_loop)

dds_loop <- DESeq(dds_loop, test="LRT", reduced = ~ 1)
resTC_loop <- DESeq2::results(dds_loop)

tab_resTC_loop <- as.data.frame(resTC_loop)
tab_resTC_loop$gene_name <- rownames(tab_resTC_loop) 
tab_resTC_loop$gene_name <- sub("XF:Z:","",tab_resTC_loop$gene_name)


tab_resTC_loop <- cbind(tab_resTC_loop,Gene_ID_Flybase[match(tab_resTC_loop$gene_name,Gene_ID_Flybase$Gene_name),"FBgn_ID"])
colnames(tab_resTC_loop)[ncol(tab_resTC_loop)] ="FBgn_ID"

top_each_time_F[[i]] <- tab_resTC_loop
}

top_F_8 <- top_each_time_F[["8"]]
top_F_24 <- top_each_time_F[["24"]]
top_F_72 <- top_each_time_F[["72"]]

#####################################################
# Loop response at each time point (males)
top_each_time_M <- vector("list")

times <- c("8", "24", "72")
for(i in times){
  colData_loop <- colData_pr %>% 
  filter(time == i & Sex == "M")
countdata_loop <- countdata %>% 
  select(colData_loop$Group)
matrix_countdata_loop <- as.matrix(countdata_loop)

dds_loop <- DESeqDataSetFromMatrix(countData = matrix_countdata_loop,
colData = colData_loop,
design = ~  Treatment)



dds_loop <- estimateSizeFactors(dds_loop)

dds_loop <- DESeq(dds_loop, test="LRT", reduced = ~ 1)
resTC_loop <- DESeq2::results(dds_loop)

tab_resTC_loop <- as.data.frame(resTC_loop)
tab_resTC_loop$gene_name <- rownames(tab_resTC_loop) 
tab_resTC_loop$gene_name <- sub("XF:Z:","",tab_resTC_loop$gene_name)


tab_resTC_loop <- cbind(tab_resTC_loop,Gene_ID_Flybase[match(tab_resTC_loop$gene_name,Gene_ID_Flybase$Gene_name),"FBgn_ID"])
colnames(tab_resTC_loop)[ncol(tab_resTC_loop)] ="FBgn_ID"

top_each_time_M[[i]] <- tab_resTC_loop
}

top_M_8 <- top_each_time_M[["8"]]
top_M_24 <- top_each_time_M[["24"]]
top_M_72 <- top_each_time_M[["72"]]


```


## WGCNA 95%
```{r}
# Uncomment and modify the following to install any missing packages
# install.packages(c("tidyverse", "magrittr", "WGCNA"))
library(tidyverse)     # tidyverse will pull in ggplot2, readr, other useful libraries
library(magrittr)      # provides the %>% operator
library(WGCNA)   
```

```{r}
library(DESeq2)

colData_pr <- colData_pr %>% 
  mutate(time_factor = factor(time, levels = c("8", "24", "72")),
         Treatment = factor(Treatment, levels = c("UC", "Pr")),
         condition = factor(paste(Sex, Treatment, time_factor, sep = "_"), levels = c("F_UC_8", "M_UC_8", "F_UC_24", "M_UC_24", "F_UC_72", "M_UC_72", "F_Pr_8", "M_Pr_8", "F_Pr_24", "M_Pr_24", "F_Pr_72", "M_Pr_72")))

countdata_pr <- countdata %>% 
  select(colData_pr$Group)
matrix_countdata_pr <- as.matrix(countdata_pr)

dds <- DESeqDataSetFromMatrix(matrix_countdata_pr,
                              colData_pr,
                              design = ~condition)
#> converting counts to integer mode
#> Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
#> design formula are characters, converting to factors

dds <- DESeq(dds)

#> estimating size factors
#> estimating dispersions
#> gene-wise dispersion estimates
#> mean-dispersion relationship
#> final dispersion estimates
#> fitting model and testing
vsd <- varianceStabilizingTransformation(dds)
library(genefilter)      # <= why is this here?
#>
#> Attaching package: 'genefilter'
#> The following objects are masked from 'package:matrixStats':
#>
#>     rowSds, rowVars
#> The following object is masked from 'package:readr':
#>
#>     spec
wpn_vsd <- getVarianceStabilizedData(dds)
rv_wpn <- rowVars(wpn_vsd)
summary(rv_wpn)
#>     Min.  1st Qu.   Median     Mean  3rd Qu.     Max.
#>  0.00000  0.00000  0.00000  0.08044  0.03322 11.14529

q75_wpn <- quantile( rowVars(wpn_vsd), .75)  # <= original
q95_wpn <- quantile( rowVars(wpn_vsd), .95)  # <= changed to 95 quantile to reduce dataset
expr_normalized <- wpn_vsd[ rv_wpn > q95_wpn, ]

expr_normalized[1:5,1:10]
#>                       B-3      B-4      B-5      L-3      L-4      L-5      S-3
#> AC149818.2_FG001 7.600901 7.077399 7.803434 7.220840 7.410408 8.028223 7.160846
#> AC149829.2_FG003 8.782014 8.179876 7.900062 8.299778 7.529891 8.631731 8.055118
#> AC182617.3_FG001 8.047244 7.120668 6.885533 7.501391 7.279413 7.809565 7.184253
#> AC186512.3_FG001 6.901539 7.389644 6.975945 6.859593 7.370816 6.633722 7.798843
#> AC186512.3_FG007 7.919688 7.754506 7.670946 7.417760 7.988427 7.904850 7.484542
#>                       S-4      S-5   B_L1.1
#> AC149818.2_FG001 7.401382 7.345322 6.524435
#> AC149829.2_FG003 8.744502 8.142909 8.240407
#> AC182617.3_FG001 8.140134 6.972400 7.777347
#> AC186512.3_FG001 6.949501 6.952659 6.059033
#> AC186512.3_FG007 8.375664 7.762799 6.335663
dim(expr_normalized)
#> [1] 5486   24

expr_normalized_df <- data.frame(expr_normalized) %>%
  mutate(
    Gene_id = row.names(expr_normalized)
  ) %>%
  pivot_longer(-Gene_id)

# expr_normalized_df %>% ggplot(., aes(x = name, y = value)) +
#   geom_violin() +
#   geom_point() +
#   theme_bw() +
#   theme(
#     axis.text.x = element_text( angle = 90)
#   ) +
#   ylim(0, NA) +
#   labs(
#     title = "Normalized and 75 quantile Expression",
#     x = "treatment",
#     y = "normalized expression"
#   )
```


```{r}
input_mat = t(expr_normalized)

input_mat[1:5,1:10]           # Look at first 5 rows and 10 columns

#library(WGCNA)
allowWGCNAThreads()          # allow multi-threading (optional)
#> Allowing multi-threading with up to 4 threads.

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to = 20, by = 2))

# Call the network topology analysis function
sft = pickSoftThreshold(
  input_mat,             # <= Input data
  #blockSize = 30,
  powerVector = powers,
  verbose = 5
  )

par(mfrow = c(1,2));
cex1 = 0.9;

plot(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     xlab = "Soft Threshold (power)",
     ylab = "Scale Free Topology Model Fit, signed R^2",
     main = paste("Scale independence")
)
text(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     labels = powers, cex = cex1, col = "red"
)
abline(h = 0.90, col = "red")
plot(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     xlab = "Soft Threshold (power)",
     ylab = "Mean Connectivity",
     type = "n",
     main = paste("Mean connectivity")
)
text(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     labels = powers,
     cex = cex1, col = "red")
```


```{r}
picked_power = 12
temp_cor <- cor       
cor <- WGCNA::cor         # Force it to use WGCNA cor function (fix a namespace conflict issue)
netwk <- blockwiseModules(input_mat,                # <= input here

                          # == Adjacency Function ==
                          power = picked_power,                # <= power here
                          networkType = "signed",

                          # == Tree and Block Options ==
                          deepSplit = 2,
                          pamRespectsDendro = F,
                          # detectCutHeight = 0.75,
                          minModuleSize = 30,
                          maxBlockSize = 4000,

                          # == Module Adjustments ==
                          reassignThreshold = 0,
                          mergeCutHeight = 0.25,

                          # == TOM == Archive the run results in TOM file (saves time)
                          saveTOMs = T,
                          saveTOMFileBase = "ER",

                          # == Output Options
                          numericLabels = T,
                          verbose = 3)


cor <- temp_cor     # Return cor function to original namespace
```


```{r}
# Convert labels to colors for plotting
mergedColors = labels2colors(netwk$colors)
# Plot the dendrogram and the module colors underneath
plotDendroAndColors(
  netwk$dendrograms[[1]],
  mergedColors[netwk$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05 )
```

```{r}
module_df_95 <- data.frame(
  gene_name = names(netwk$colors),
  colors = labels2colors(netwk$colors)
)

module_df_95[1:5,]
#>            gene_id    colors
#> 1 AC149818.2_FG001      blue
#> 2 AC149829.2_FG003      blue
#> 3 AC182617.3_FG001      blue
#> 4 AC186512.3_FG001 turquoise
#> 5 AC186512.3_FG007 turquoise

# write_delim(module_df,
#            file = "Results/gene_modules.txt",
#            delim = "\t")
```


```{r}
# Get Module Eigengenes per cluster
MEs0 <- moduleEigengenes(input_mat, mergedColors)$eigengenes

# Reorder modules so similar modules are next to each other
MEs0 <- orderMEs(MEs0)
module_order = names(MEs0) %>% gsub("ME","", .)

# Add treatment names
MEs0$treatment = row.names(MEs0)

# tidy & plot data
mME = MEs0 %>%
  pivot_longer(-treatment) %>%
  mutate(
    name = gsub("ME", "", name),
    name = factor(name, levels = module_order)
  ) %>% 
  filter(name %in% c("yellow"))

mME$treatment = factor(mME$treatment, levels = c("Y_F_UC_8_1", "Y_F_UC_8_2", "Y_F_UC_8_3",
                                   "Y_M_UC_8_1", "Y_M_UC_8_2", "Y_M_UC_8_3", 
                                   "Y_F_Pr_8_1", "Y_F_Pr_8_2", "Y_F_Pr_8_3",
                                   "Y_M_Pr_8_1", "Y_M_Pr_8_2", "Y_M_Pr_8_3",
                                   "Y_F_UC_24_1", "Y_F_UC_24_2", "Y_F_UC_24_3",
                                   "Y_M_UC_24_1", "Y_M_UC_24_2", "Y_M_UC_24_3", 
                                   "Y_F_Pr_24_1", "Y_F_Pr_24_2", "Y_F_Pr_24_3",
                                   "Y_M_Pr_24_1", "Y_M_Pr_24_2", "Y_M_Pr_24_3",
                                   "Y_F_UC_72_1", "Y_F_UC_72_2", "Y_F_UC_72_3",
                                   "Y_M_UC_72_1", "Y_M_UC_72_2", "Y_M_UC_72_3", 
                                   "Y_F_Pr_72_1", "Y_F_Pr_72_2", "Y_F_Pr_72_3",
                                   "Y_M_Pr_72_1", "Y_M_Pr_72_2", "Y_M_Pr_72_3"))

mME <- mME %>% 
  separate(treatment, into = c("Dissected", "Sex", "Treatment", "Time", "Rep"), remove = FALSE)


(module_heatmap <- mME %>% ggplot(., aes(x=treatment, y=name, fill=value)) +
  geom_tile() +
  theme_bw() +
  scale_fill_viridis() +
  theme(axis.text.x = element_text(angle=90)) +
  labs(title = "Module-trait Relationships", y = "Modules", fill="corr") +
  annotate("segment", x = 0, xend = 6.5, y = 1.7, yend = 1.7, colour = "#000004", linewidth = 5) +
  annotate("segment", x = 6.5, xend = 12.5, y = 1.7, yend = 1.7, colour = "#FEC287", linewidth = 5) +
  annotate("segment", x = 12.5, xend = 18.5, y = 1.7, yend = 1.7, colour = "#000004", linewidth = 5) +
  annotate("segment", x = 18.5, xend = 24.5, y = 1.7, yend = 1.7, colour = "#F56B5C", linewidth = 5) +
  annotate("segment", x = 24.5, xend = 30.5, y = 1.7, yend = 1.7, colour = "#000004", linewidth = 5) +
  annotate("segment", x = 30.5, xend = 37, y = 1.7, yend = 1.7, colour = "#A02F7F", linewidth = 5) +
  annotate("segment", x = 0, xend = 3.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 3.5, xend = 6.5, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  annotate("segment", x = 6.5, xend = 9.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 9.5, xend = 12.5, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  annotate("segment", x = 12.5, xend = 15.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 15.5, xend = 18.5, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  annotate("segment", x = 18.5, xend = 21.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 21.5, xend = 24.5, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  annotate("segment", x = 24.5, xend = 27.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 27.5, xend = 30.5, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  annotate("segment", x = 30.5, xend = 33.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 33.5, xend = 37, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  expand_limits(y = 1.75))
#ggsave(filename = "Results/DESeq/module_heatmap.png", module_heatmap, width = 6.5, height = 3)
```


```{r}
# pick out a few modules of interest here
modules_of_interest = "yellow"

# Pull out list of genes in that module
submod = module_df_95 %>%
  subset(colors %in% modules_of_interest)

row.names(module_df_95) = module_df_95$gene_name

# Get normalized expression for those genes
expr_normalized[1:5,1:10]

subexpr = expr_normalized[submod$gene_name,]

submod_df = data.frame(subexpr) %>%
  mutate(
    gene_name = row.names(.)
  ) %>%
  pivot_longer(-gene_name) %>%
  mutate(
    module = module_df_95[gene_name,]$colors,
    )

submod_df$name = factor(submod_df$name, levels = c("Y_F_UC_8_1", "Y_F_UC_8_2", "Y_F_UC_8_3",
                                   "Y_M_UC_8_1", "Y_M_UC_8_2", "Y_M_UC_8_3", 
                                   "Y_F_Pr_8_1", "Y_F_Pr_8_2", "Y_F_Pr_8_3",
                                   "Y_M_Pr_8_1", "Y_M_Pr_8_2", "Y_M_Pr_8_3",
                                   "Y_F_UC_24_1", "Y_F_UC_24_2", "Y_F_UC_24_3",
                                   "Y_M_UC_24_1", "Y_M_UC_24_2", "Y_M_UC_24_3", 
                                   "Y_F_Pr_24_1", "Y_F_Pr_24_2", "Y_F_Pr_24_3",
                                   "Y_M_Pr_24_1", "Y_M_Pr_24_2", "Y_M_Pr_24_3",
                                   "Y_F_UC_72_1", "Y_F_UC_72_2", "Y_F_UC_72_3",
                                   "Y_M_UC_72_1", "Y_M_UC_72_2", "Y_M_UC_72_3", 
                                   "Y_F_Pr_72_1", "Y_F_Pr_72_2", "Y_F_Pr_72_3",
                                   "Y_M_Pr_72_1", "Y_M_Pr_72_2", "Y_M_Pr_72_3"))

(expression_plot <- submod_df %>% ggplot(., aes(x=name, y=value, group=gene_name)) +
  geom_line(aes(color = module),
            alpha = 0.2) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90)
  ) +
  facet_grid(rows = vars(module)) +
  scale_color_manual(values = "#DBA800") +
  labs(x = "treatment",
       y = "normalized expression"))
#ggsave("Results/DESeq/expression_module.png", expression_plot, width = 6.5, height = 3)
```

```{r}
genes_of_interest = module_df_95 %>%
  subset(colors %in% modules_of_interest)

expr_of_interest = expr_normalized[genes_of_interest$gene_name,]
expr_of_interest[1:5,1:5]


# Only recalculate TOM for modules of interest (faster, altho there's some online discussion if this will be slightly off)
TOM = TOMsimilarityFromExpr(t(expr_of_interest),
                            power = picked_power)
#> TOM calculation: adjacency..
#> ..will use 4 parallel threads.
#>  Fraction of slow calculations: 0.000000
#> ..connectivity..
#> ..matrix multiplication (system BLAS)..
#> ..normalization..
#> ..done.

# Add gene names to row and columns
row.names(TOM) = row.names(expr_of_interest)
colnames(TOM) = row.names(expr_of_interest)

edge_list = data.frame(TOM) %>%
  mutate(
    gene1 = row.names(.)
  ) %>%
  pivot_longer(-gene1) %>%
  dplyr::rename(gene2 = name, correlation = value) %>%
  unique() %>%
  subset(!(gene1==gene2)) %>%
  mutate(
    module1 = module_df_95[gene1,]$colors,
    module2 = module_df_95[gene2,]$colors
  )

head(edge_list)

```


# Responses ignoring sex
## Volcano plot
```{r}
mutate <- dplyr::mutate
volcano_plots <- vector("list")
times <- c("8", "24", "72")
colors_vol <- vector("list")
colors_vol[["8"]] <- c("#000004", "lightgrey", "#FEC287")
colors_vol[["24"]] <- c("#000004", "lightgrey", "#F56B5C")
colors_vol[["72"]] <- c("#000004", "lightgrey", "#A02F7F")
for(i in times){
  
library(ggrepel)
top <- top_each_time[[i]]
top$gene_name <- rownames(top)


top_vol <- top %>% 
  mutate(color = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                           log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                           TRUE ~ "NS"))
colors <- colors_vol[[i]]

(time_volcano <- top_vol %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = color)) +
  geom_text_repel(data = top_vol %>% filter(!color == "NS"), aes(label = gene_name), max.overlaps = 5) +
  scale_color_manual(values = colors, name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab(paste("Log2 Fold Change at", i, "hours")) +
  ylab("-log10(P value)") +
  ggtitle(paste("Genes differentially expressed\nin carcasses during P. rettgeri infection", "\n(", i, " hours post injection)", sep = "")))
ggsave(filename = paste("Results/DESeq/", i, "_volcano.png", sep = ""), time_volcano, width = 4.5, height = 4)
}

```

## Time ClusterProfiler GSEA

```{r}
ids<-bitr(rownames(top_8), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

df2 = top_8[rownames(top_8) %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_8 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

require(DOSE)

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(kegg_8_dotplot <- dotplot(kegg_8, showCategory=10, split=".sign", font.size = 7) +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched gene sets in Drosophila 8 hours\npost P. rettgeri injection"))
ggsave("Results/DESeq/kegg_8_dotplot.png", plot = kegg_8_dotplot, width = 6.5, height = 4) 


#################################################
# 24 hours
ids<-bitr(top_24$FBgn_ID, fromType = "FLYBASE", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("FLYBASE")]),]

df2 = top_24[top_24$FBgn_ID %in% dedup_ids$FLYBASE,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_24 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")


direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(kegg_24_dotplot <- dotplot(kegg_24, showCategory=10, split=".sign", font.size = 10) +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched gene sets in Drosophila 24\nhours post P. rettgeri injection"))
#ggsave("Results/DESeq/kegg_24_dotplot.png", plot = kegg_24_dotplot, width = 6.5, height = 4) 



#################################################
# 72 hours
ids<-bitr(top_72$FBgn_ID, fromType = "FLYBASE", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("FLYBASE")]),]

df2 = top_72[top_72$FBgn_ID %in% dedup_ids$FLYBASE,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_72 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")


direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(kegg_72_dotplot <- dotplot(kegg_72, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched gene sets in Drosophila\n72 hours post P. rettgeri injection"))
#ggsave("Results/DESeq/kegg_72_dotplot.png", plot = kegg_72_dotplot, width = 6.5, height = 4) 
```

## Venn diagram
```{r}
# Load library
library(VennDiagram)
colors_time <- c("#FEC287", "#F56B5C", "#A02F7F")
 
top_8_genes_upregulated <- top_each_time$`8` %>% 
  filter(pvalue < 0.05 & log2FoldChange > 0)
top_24_genes_upregulated <- top_each_time$`24` %>% 
  filter(pvalue < 0.05 & log2FoldChange > 0)
top_72_genes_upregulated <- top_each_time$`72` %>% 
  filter(pvalue < 0.05 & log2FoldChange > 0)
top_8_upreg_genes <- rownames(top_8_genes_upregulated)
top_24_upreg_genes <- rownames(top_24_genes_upregulated)
top_72_upreg_genes <- rownames(top_72_genes_upregulated)

upreg_venn <- venn.diagram(x = list(top_8_upreg_genes, top_24_upreg_genes, top_72_upreg_genes),
             category.names = c("Upregulated at 8 hours", "Upregulated at 24 hours", "Upregulated at 72 hours"),
             fill = (alpha(colors_time, 0.5)), 
             col = colors_time,
             alpha = 0.5,
             main="Genes upregulated during infection", filename = NULL,
             main.fontface = "bold",
             main.fontfamily = "sans",
             cat.fontfamily= "sans",
             margin = 0.125,
             cat.cex = 0.75,
             cat.dist = 0.075)
#ggsave(upreg_venn, filename = "Results/upreg_venn.png", width = 6, height = 6)


top_8_genes_downregulated <- top_each_time$`8` %>% 
  filter(pvalue < 0.05 & log2FoldChange < 0)
top_24_genes_downregulated <- top_each_time$`24` %>% 
  filter(pvalue < 0.05 & log2FoldChange < 0)
top_72_genes_downregulated <- top_each_time$`72` %>% 
  filter(pvalue < 0.05 & log2FoldChange < 0)
top_8_downreg_genes <- rownames(top_8_genes_downregulated)
top_24_downreg_genes <- rownames(top_24_genes_downregulated)
top_72_downreg_genes <- rownames(top_72_genes_downregulated)

downreg_venn <- venn.diagram(x = list(top_8_downreg_genes, top_24_downreg_genes, top_72_downreg_genes),
             category.names = c("Downregulated at 8 hours", "Downregulated at 24 hours", "Downregulated at 72 hours"),
             fill = (alpha(colors_time, 0.5)), 
             col = colors_time,
             alpha = 0.5,
             main="Genes downregulated during infection", filename = NULL,
             main.fontface = "bold",
             main.fontfamily = "sans",
             cat.fontfamily= "sans",
             margin = 0.125,
             cat.cex = 0.75,
             cat.dist = 0.075)
#ggsave(downreg_venn, filename = "Results/downreg_venn.png", width = 6, height = 6)
```


### Female
```{r}
# Load library
library(VennDiagram)
colors_time <- c("#FEC287", "#F56B5C", "#A02F7F")
 
top_F_8_genes_upregulated <- top_each_time_F$`8` %>% 
  filter(pvalue < 0.05 & log2FoldChange > 0)
top_F_24_genes_upregulated <- top_each_time_F$`24` %>% 
  filter(pvalue < 0.05 & log2FoldChange > 0)
top_F_72_genes_upregulated <- top_each_time_F$`72` %>% 
  filter(pvalue < 0.05 & log2FoldChange > 0)
top_F_8_upreg_genes <- rownames(top_F_8_genes_upregulated)
top_F_24_upreg_genes <- rownames(top_F_24_genes_upregulated)
top_F_72_upreg_genes <- rownames(top_F_72_genes_upregulated)

upreg_venn_F <- venn.diagram(x = list(top_F_8_upreg_genes, top_F_24_upreg_genes, top_F_72_upreg_genes),
             category.names = c("Upregulated at 8 hours", "Upregulated at 24 hours", "Upregulated at 72 hour"),
             fill = (alpha(colors_time, 0.5)), 
             col = colors_time,
             alpha = 0.5,
             main="Genes upregulated during infection (females)", filename = NULL,
             main.fontface = "bold",
             main.fontfamily = "sans",
             cat.fontfamily= "sans",
             margin = 0.125,
             cat.cex = 0.75,
             cat.dist = 0.075)
#ggsave(upreg_venn_F, filename = "Results/upreg_venn_F.png", width = 8, height = 3)


top_F_8_genes_downregulated <- top_each_time_F$`8` %>% 
  filter(pvalue < 0.05 & log2FoldChange < 0)
top_F_24_genes_downregulated <- top_each_time_F$`24` %>% 
  filter(pvalue < 0.05 & log2FoldChange < 0)
top_F_72_genes_downregulated <- top_each_time_F$`72` %>% 
  filter(pvalue < 0.05 & log2FoldChange < 0)
top_F_8_downreg_genes <- rownames(top_F_8_genes_downregulated)
top_F_24_downreg_genes <- rownames(top_F_24_genes_downregulated)
top_F_72_downreg_genes <- rownames(top_F_72_genes_downregulated)

downreg_venn_F <- venn.diagram(x = list(top_F_8_downreg_genes, top_F_24_downreg_genes, top_F_72_downreg_genes),
             category.names = c("Downregulated at 8 hours", "Downregulated at 24 hours", "Downregulated at 72 hours"),
             fill = (alpha(colors_time, 0.5)), 
             col = colors_time,
             alpha = 0.5,
             main="Genes downregulated during infection (females)", filename = NULL,
             main.fontface = "bold",
             main.fontfamily = "sans",
             cat.fontfamily= "sans",
             margin = 0.125,
             cat.cex = 0.75,
             cat.dist = 0.075)
#ggsave(downreg_venn_F, filename = "Results/downreg_venn_F.png", width = 6, height = 6)
```

### Male
```{r}
# Load library
library(VennDiagram)
colors_time <- c("#FEC287", "#F56B5C", "#A02F7F")
 
top_M_8_genes_upregulated <- top_each_time_M$`8` %>% 
  filter(pvalue < 0.05 & log2FoldChange > 0)
top_M_24_genes_upregulated <- top_each_time_M$`24` %>% 
  filter(pvalue < 0.05 & log2FoldChange > 0)
top_M_72_genes_upregulated <- top_each_time_M$`72` %>% 
  filter(pvalue < 0.05 & log2FoldChange > 0)
top_M_8_upreg_genes <- rownames(top_M_8_genes_upregulated)
top_M_24_upreg_genes <- rownames(top_M_24_genes_upregulated)
top_M_72_upreg_genes <- rownames(top_M_72_genes_upregulated)

upreg_venn_M <- venn.diagram(x = list(top_M_8_upreg_genes, top_M_24_upreg_genes, top_M_72_upreg_genes),
             category.names = c("Upregulated at 8 hours", "Upregulated at 24 hours", "Upregulated at 72 hour"),
             fill = (alpha(colors_time, 0.5)), 
             col = colors_time,
             alpha = 0.5,
             main="Genes upregulated during infection (males)", filename = NULL,
             main.fontface = "bold",
             main.fontfamily = "sans",
             cat.fontfamily= "sans",
             margin = 0.125,
             cat.cex = 0.75,
             cat.dist = 0.075)
#ggsave(upreg_venn_M, filename = "Results/upreg_venn_M.png", width = 8, height = 3)


top_M_8_genes_downregulated <- top_each_time_M$`8` %>% 
  filter(pvalue < 0.05 & log2FoldChange < 0)
top_M_24_genes_downregulated <- top_each_time_M$`24` %>% 
  filter(pvalue < 0.05 & log2FoldChange < 0)
top_M_72_genes_downregulated <- top_each_time_M$`72` %>% 
  filter(pvalue < 0.05 & log2FoldChange < 0)
top_M_8_downreg_genes <- rownames(top_M_8_genes_downregulated)
top_M_24_downreg_genes <- rownames(top_M_24_genes_downregulated)
top_M_72_downreg_genes <- rownames(top_M_72_genes_downregulated)

downreg_venn_M <- venn.diagram(x = list(top_M_8_downreg_genes, top_M_24_downreg_genes, top_M_72_downreg_genes),
             category.names = c("Downregulated at 8 hours", "Downregulated at 24 hours", "Downregulated at 72 hours"),
             fill = (alpha(colors_time, 0.5)), 
             col = colors_time,
             alpha = 0.5,
             main="Genes downregulated during infection (males)", filename = NULL,
             main.fontface = "bold",
             main.fontfamily = "sans",
             cat.fontfamily= "sans",
             margin = 0.125,
             cat.cex = 0.75,
             cat.dist = 0.075)
#ggsave(downreg_venn_M, filename = "Results/downreg_venn_M.png", width = 6, height = 6)
```



### Upregulated shared genes
```{r}
shared_upreg <- top_each_time$`24` %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(gene_name %in% top_8_upreg_genes & gene_name %in% top_24_upreg_genes)

all_upreg <- shared_upreg %>% 
  filter(gene_name %in% top_72_upreg_genes)
up_8_24 <- shared_upreg %>% 
  filter(!gene_name %in% top_72_upreg_genes)


original_gene_list <- shared_upreg$log2FoldChange
names(original_gene_list) <- shared_upreg$gene_name

ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

gene <- dedup_ids$ENTREZID

upreg_both_kegg <- enrichKEGG(gene = gene,
                 organism = "dme",
                 keyType = "ncbi-geneid",
                 pvalueCutoff = 0.05)

dotplot(upreg_both_kegg)


parallel_upreg_both <- countdata_dis %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(data_colnames), names_to = "Sample", values_to = "Count") %>% 
  #left_join(top_time) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  filter(gene_name %in% up_8_24$gene_name) %>% 
  mutate(Challenge = factor(Challenge, levels = c("UC", "Pr"))) %>% 
  group_by(gene_name, Sex, Time, Challenge) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
   mutate(Time_factor = factor(Time),
         Sex = factor(Sex),
         gene_sample = paste(gene_name, Sex, Time, sep = "_"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")
time_labs <- c("8 hours", "24 hours", "72 hours")
names(time_labs) <- c("8", "24", "72")

(upreg_both_parallel_plot <- ggplot(parallel_upreg_both, aes(x = Challenge, y = avg_count, group = gene_sample)) +
  geom_line(aes(color = Time_factor)) +
  geom_text_repel(data = (parallel_upreg_both %>% dplyr::distinct(gene_sample, .keep_all = TRUE)),aes(x = Challenge, y = avg_count,label = gene_name)) +
  facet_wrap(Sex~Time_factor, labeller = labeller(Sex = sex_labs, Time_factor = time_labs)) +
  scale_color_manual(values = colors_time) +
  theme_bw() +
  ylab("Gene count") +
  theme(aspect.ratio = 1, legend.position = "none", strip.background = element_rect(fill = "white")) +
  ggtitle("Expression of genes significantly upregated at both 8 and 24 hours\n after P. rettgeri injection"))
#ggsave(filename = "Results/upreg_both_parallel_plot.png", upreg_both_parallel_plot, width= 12, height = 8)

```

### Downregulated shared genes
```{r}
shared_downreg <- top_each_time$`24` %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(gene_name %in% top_8_downreg_genes & gene_name %in% top_24_downreg_genes)

all_downreg <- shared_downreg %>% 
  filter(gene_name %in% top_72_downreg_genes)
down_8_24 <- shared_downreg %>% 
  filter(!gene_name %in% top_72_downreg_genes)
down_24_72 <- top_each_time$`24` %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(!gene_name %in% top_8_downreg_genes & gene_name %in% top_24_downreg_genes & gene_name %in% top_72_downreg_genes)


parallel_downreg_both <- countdata_dis %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(data_colnames), names_to = "Sample", values_to = "Count") %>% 
  #left_join(top_time) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  filter(gene_name %in% down_8_24$gene_name | gene_name %in% down_24_72$gene_name | gene_name %in% all_downreg$gene_name) %>% 
  mutate(Challenge = factor(Challenge, levels = c("UC", "Pr"))) %>% 
  group_by(gene_name, Sex, Time, Challenge) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
   mutate(Time_factor = factor(Time),
         Sex = factor(Sex),
         gene_sample = paste(gene_name, Sex, Time, sep = "_"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")
time_labs <- c("8 hours", "24 hours", "72 hours")
names(time_labs) <- c("8", "24", "72")

(downreg_both_parallel_plot <- ggplot(parallel_downreg_both, aes(x = Challenge, y = avg_count, group = gene_sample)) +
  geom_line(aes(color = Time_factor)) +
  geom_point(aes(y = Count, color = Time_factor)) +
  geom_text_repel(data = (parallel_downreg_both %>% filter(Challenge == "UC") %>% dplyr::distinct(gene_sample, .keep_all = TRUE)),aes(x = Challenge, y = avg_count,label = gene_name)) +
  facet_wrap(Sex~Time_factor, labeller = labeller(Sex = sex_labs, Time_factor = time_labs)) +
  scale_color_manual(values = colors_time) +
  theme_bw() +
  ylab("Gene count") +
  theme(aspect.ratio = 1, legend.position = "none", strip.background = element_rect(fill = "white")) +
  ggtitle("Expression of genes significantly downregulated at both 8 and 24 hours\n after P. rettgeri injection"))
#ggsave(filename = "Results/downreg_all_parallel_plot.png", downreg_both_parallel_plot, width= 8, height = 6)

```


### Acute response
#### WGCNA
```{r}
countdata_acute <- countdata %>% 
  select(contains("8")) %>% 
  select(!contains("WD"))

col_sel <- names(countdata_acute)

my_data <- countdata_acute %>% 
  tidyr::pivot_longer(
    .,
    col = all_of(col_sel)
  ) %>% 
  mutate(group = name)

#==== Plot groups (Sample Groups vs RNA Seq Counts) to identify outliers
# (
#  p <- my_data %>%
#     ggplot(., aes(x = name, y = value)) +             # x = treatment, y = RNA Seq count
#     geom_violin() +                                   # violin plot, show distribution
#     geom_point(alpha = 0.2) +                         # scatter plot
#     theme_bw() +
#     theme(
#       axis.text.x = element_text(angle = 90)          # Rotate treatment text
#     ) +
#     labs(x = "Treatment Groups", y = "RNA Seq Counts") +
#     facet_grid(cols = vars(group), drop = TRUE, scales = "free_x")      # Facet by hour
# )
```

```{r}
library(DESeq2)

colData_acute <- colData %>% 
  filter(Treatment != "WD" & time == 8) %>% 
  mutate(Treatment = factor(Treatment, levels = c("UC", "Pr")),
         condition = factor(paste(Sex, Treatment, sep = "_"), levels = c("F_UC", "M_UC","F_Pr", "M_Pr")))

matrix_countdata_acute <- as.matrix(countdata_acute)

dds <- DESeqDataSetFromMatrix(matrix_countdata_acute,
                              colData_acute,
                              design = ~condition)
#> converting counts to integer mode
#> Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
#> design formula are characters, converting to factors

dds <- DESeq(dds)

#> estimating size factors
#> estimating dispersions
#> gene-wise dispersion estimates
#> mean-dispersion relationship
#> final dispersion estimates
#> fitting model and testing
vsd <- varianceStabilizingTransformation(dds)
library(genefilter)      # <= why is this here?
#>
#> Attaching package: 'genefilter'
#> The following objects are masked from 'package:matrixStats':
#>
#>     rowSds, rowVars
#> The following object is masked from 'package:readr':
#>
#>     spec
wpn_vsd <- getVarianceStabilizedData(dds)
rv_wpn <- rowVars(wpn_vsd)
summary(rv_wpn)
#>     Min.  1st Qu.   Median     Mean  3rd Qu.     Max.
#>  0.00000  0.00000  0.00000  0.08044  0.03322 11.14529

q75_wpn <- quantile( rowVars(wpn_vsd), .75)  # <= original
q95_wpn <- quantile( rowVars(wpn_vsd), .95)  # <= changed to 95 quantile to reduce dataset
expr_normalized <- wpn_vsd[ rv_wpn > q95_wpn, ]

expr_normalized[1:5,1:10]
#>                       B-3      B-4      B-5      L-3      L-4      L-5      S-3
#> AC149818.2_FG001 7.600901 7.077399 7.803434 7.220840 7.410408 8.028223 7.160846
#> AC149829.2_FG003 8.782014 8.179876 7.900062 8.299778 7.529891 8.631731 8.055118
#> AC182617.3_FG001 8.047244 7.120668 6.885533 7.501391 7.279413 7.809565 7.184253
#> AC186512.3_FG001 6.901539 7.389644 6.975945 6.859593 7.370816 6.633722 7.798843
#> AC186512.3_FG007 7.919688 7.754506 7.670946 7.417760 7.988427 7.904850 7.484542
#>                       S-4      S-5   B_L1.1
#> AC149818.2_FG001 7.401382 7.345322 6.524435
#> AC149829.2_FG003 8.744502 8.142909 8.240407
#> AC182617.3_FG001 8.140134 6.972400 7.777347
#> AC186512.3_FG001 6.949501 6.952659 6.059033
#> AC186512.3_FG007 8.375664 7.762799 6.335663
dim(expr_normalized)
#> [1] 5486   24

expr_normalized_df <- data.frame(expr_normalized) %>%
  mutate(
    Gene_id = row.names(expr_normalized)
  ) %>%
  pivot_longer(-Gene_id)

# expr_normalized_df %>% ggplot(., aes(x = name, y = value)) +
#   geom_violin() +
#   geom_point() +
#   theme_bw() +
#   theme(
#     axis.text.x = element_text( angle = 90)
#   ) +
#   ylim(0, NA) +
#   labs(
#     title = "Normalized and 95 quantile Expression",
#     x = "treatment",
#     y = "normalized expression"
#   )
```


```{r}
input_mat = t(expr_normalized)

input_mat[1:5,1:10]           # Look at first 5 rows and 10 columns

#library(WGCNA)
allowWGCNAThreads()          # allow multi-threading (optional)
#> Allowing multi-threading with up to 4 threads.

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to = 20, by = 2))

# Call the network topology analysis function
sft = pickSoftThreshold(
  input_mat,             # <= Input data
  #blockSize = 30,
  powerVector = powers,
  verbose = 5
  )

par(mfrow = c(1,2));
cex1 = 0.9;

plot(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     xlab = "Soft Threshold (power)",
     ylab = "Scale Free Topology Model Fit, signed R^2",
     main = paste("Scale independence")
)
text(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     labels = powers, cex = cex1, col = "red"
)
abline(h = 0.90, col = "red")
plot(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     xlab = "Soft Threshold (power)",
     ylab = "Mean Connectivity",
     type = "n",
     main = paste("Mean connectivity")
)
text(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     labels = powers,
     cex = cex1, col = "red")
```


```{r}
picked_power = 16
temp_cor <- cor       
cor <- WGCNA::cor         # Force it to use WGCNA cor function (fix a namespace conflict issue)
netwk <- blockwiseModules(input_mat,                # <= input here

                          # == Adjacency Function ==
                          power = picked_power,                # <= power here
                          networkType = "signed",

                          # == Tree and Block Options ==
                          deepSplit = 2,
                          pamRespectsDendro = F,
                          # detectCutHeight = 0.75,
                          minModuleSize = 30,
                          maxBlockSize = 4000,

                          # == Module Adjustments ==
                          reassignThreshold = 0,
                          mergeCutHeight = 0.25,

                          # == TOM == Archive the run results in TOM file (saves time)
                          saveTOMs = T,
                          saveTOMFileBase = "ER",

                          # == Output Options
                          numericLabels = T,
                          verbose = 3)


cor <- temp_cor     # Return cor function to original namespace
```


```{r}
# Convert labels to colors for plotting
mergedColors = labels2colors(netwk$colors)
# Plot the dendrogram and the module colors underneath
plotDendroAndColors(
  netwk$dendrograms[[1]],
  mergedColors[netwk$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05 )
```

```{r}
acute_module_df_95 <- data.frame(
  gene_name = names(netwk$colors),
  colors = labels2colors(netwk$colors)
)

acute_module_df_95[1:5,]
#>            gene_id    colors
#> 1 AC149818.2_FG001      blue
#> 2 AC149829.2_FG003      blue
#> 3 AC182617.3_FG001      blue
#> 4 AC186512.3_FG001 turquoise
#> 5 AC186512.3_FG007 turquoise

# write_delim(module_df,
#            file = "Results/gene_modules.txt",
#            delim = "\t")
```


```{r}
# Get Module Eigengenes per cluster
MEs0 <- moduleEigengenes(input_mat, mergedColors)$eigengenes

# Reorder modules so similar modules are next to each other
MEs0 <- orderMEs(MEs0)
module_order = names(MEs0) %>% gsub("ME","", .)

# Add treatment names
MEs0$treatment = row.names(MEs0)

modules_of_interest = c("blue", "turquoise", "grey")

# tidy & plot data
mME = MEs0 %>%
  pivot_longer(-treatment) %>%
  mutate(
    name = gsub("ME", "", name),
    name = factor(name, levels = module_order)
  ) %>% 
  filter(name %in% modules_of_interest)

(module_heatmap <- mME %>% ggplot(., aes(x=treatment, y=name, fill=value)) +
  geom_tile() +
  theme_bw() +
  scale_fill_viridis() +
  theme(axis.text.x = element_text(angle=90)) +
  labs(title = "Module-trait Relationships", y = "Modules", fill="corr"))
#ggsave(filename = "Results/acute/module_heatmap.png", module_heatmap, width = 12, height = 4)
```


```{r}
# pick out a few modules of interest here

# Pull out list of genes in that module
submod = acute_module_df_95 %>%
  subset(colors %in% modules_of_interest)

row.names(acute_module_df_95) = acute_module_df_95$gene_name

# Get normalized expression for those genes
expr_normalized[1:5,1:10]

subexpr = expr_normalized[submod$gene_name,]

submod_df = data.frame(subexpr) %>%
  mutate(
    gene_name = row.names(.)
  ) %>%
  pivot_longer(-gene_name) %>%
  mutate(
    module = acute_module_df_95[gene_name,]$colors,
    module = factor(module, levels = rev(module_order))
  )

(expression_plot <- submod_df %>% ggplot(., aes(x=name, y=value, group=gene_name)) +
  geom_line(aes(color = module),
            alpha = 0.2) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90)
  ) +
  facet_grid(rows = vars(module), scales = "free") +
  scale_color_manual(values = c("grey", "turquoise", "blue")) +
  labs(x = "treatment",
       y = "normalized expression"))
#ggsave("Results/acute/expression_module.png", expression_plot, width = 8, height = 6)
```

```{r}
genes_of_interest = acute_module_df_95 %>%
  subset(colors %in% modules_of_interest)

expr_of_interest = expr_normalized[genes_of_interest$gene_name,]
expr_of_interest[1:5,1:5]


# Only recalculate TOM for modules of interest (faster, altho there's some online discussion if this will be slightly off)
TOM = TOMsimilarityFromExpr(t(expr_of_interest),
                            power = picked_power)
#> TOM calculation: adjacency..
#> ..will use 4 parallel threads.
#>  Fraction of slow calculations: 0.000000
#> ..connectivity..
#> ..matrix multiplication (system BLAS)..
#> ..normalization..
#> ..done.

# Add gene names to row and columns
row.names(TOM) = row.names(expr_of_interest)
colnames(TOM) = row.names(expr_of_interest)

edge_list = data.frame(TOM) %>%
  mutate(
    gene1 = row.names(.)
  ) %>%
  pivot_longer(-gene1) %>%
  dplyr::rename(gene2 = name, correlation = value) %>%
  unique() %>%
  subset(!(gene1==gene2)) %>%
  mutate(
    module1 = acute_module_df_95[gene1,]$colors,
    module2 = acute_module_df_95[gene2,]$colors
  )

head(edge_list)

# write_delim(edge_list,
#              file = "Results/acute/edgelist.csv",
#              delim = ",")
```

##### Characterizing modules
```{r}
acute_wgcna_kegg <- vector("list")


for(i in modules_of_interest){
  genes_df <- acute_module_df_95 %>% 
    filter(colors == i)
  original_gene_list <- genes_df$colors
names(original_gene_list) <- genes_df$gene_name

ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

gene <- dedup_ids$ENTREZID

kk <- enrichKEGG(gene = gene,
                 organism = "dme",
                 keyType = "ncbi-geneid",
                 pvalueCutoff = 0.05)
acute_wgcna_kegg[[i]] <- kk


}

(kegg_acute_brown <- dotplot(acute_wgcna_kegg$grey, showCategory=10) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in grey module"))
#ggsave("Results/acute/kegg_acute_brown.png", plot = kegg_acute_brown, width = 8, height = 6) 
(kegg_acute_blue <- dotplot(acute_wgcna_kegg$blue, showCategory=10) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in blue module"))
#ggsave("Results/acute/kegg_acute_blue.png", plot = kegg_acute_blue, width = 8, height = 6) 
(kegg_acute_turquoise <- dotplot(acute_wgcna_kegg$turquoise, showCategory=10) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in turquoise module"))
#ggsave("Results/acute/kegg_acute_turquoise.png", plot = kegg_acute_turquoise, width = 8, height = 6) 


```



```{r}
acute <- top_8 %>% 
  filter(abs(log2FoldChange) >1.5 & pvalue < 0.05) %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 ~ "Upregulated",
                            log2FoldChange < -1.5 ~ "Downregulated"))
acute_genes <- rownames(acute)

(acute_volcano_8 <- acute %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "#FEC287"), name = NULL) +
  theme_bw() +
  xlab("Log2 Fold Change at 8 hours") +
  ylab("-log10(P value)") +
  ggtitle("Acute response genes differentially expressed 8 hours after P. rettgeri injection"))

colors_time <- c("#A02F7F", "#F56B5C", "#FEC287", "#000004")
acute_24 <- top_24 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% acute_genes)

(acute_volcano_24 <- acute_24 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#F56B5C"), name = NULL) +
  theme_bw() +
  xlab("Log2 Fold Change at 24 hours") +
  ylab("-log10(P value)") +
  ggtitle("Acute response genes differentially expressed 24 hours after P. rettgeri injection"))

acute_72 <- top_72 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% acute_genes)

(acute_volcano_72 <- acute_72 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("lightgrey", "#A02F7F"), name = NULL) +
  theme_bw() +
  xlab("Log2 Fold Change at 72 hours") +
  ylab("-log10(P value)") +
  ggtitle("Acute response genes differentially expressed 72 hours after P. rettgeri injection"))

```


#### Females
```{r}
acute_F <- top_F_8 %>% 
  filter(abs(log2FoldChange) >1.5 & pvalue < 0.05) %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 ~ "Upregulated",
                            log2FoldChange < -1.5 ~ "Downregulated",
                            TRUE ~ "NS"))
acute_genes <- rownames(acute_F)
color_8 <- "#FEC287"
color_24 <- "#F56B5C"
color_72 <- "#A02F7F"

(acute_volcano_F_8 <- acute %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name), max.overlaps = 2) +
  scale_color_manual(values = c("#000004", "lightgrey", "#FEC287"), name = NULL) +
  theme_bw() +
  xlab("Log2 Fold Change at 8 hours") +
  ylab("-log10(P value)") +
  theme(aspect.ratio = 1) +
  ggtitle("Differentially expressed genes 8 hours \nafter P. rettgeri injection in females"))

colors_time <- c("#A02F7F", "#F56B5C", "#FEC287", "#000004")
acute_F_24 <- top_F_24 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% acute_genes)

(acute_volcano_F_24 <- acute_F_24 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#F56B5C"), name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log2 Fold Change at 24 hours") +
  ylab("-log10(P value)") +
  ggtitle("Acute response genes differentially expressed 24 hours \nafter P. rettgeri injection in females"))

acute_F_72 <- top_F_72 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% acute_genes)

(acute_volcano_F_72 <- acute_F_72 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#A02F7F"), name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log2 Fold Change at 72 hours") +
  ylab("-log10(P value)") +
  ggtitle("Acute response genes differentially expressed 72 hours \nafter P. rettgeri injection in females"))

# ggsave("Results/acute_8hr_F_vol.png", plot = acute_volcano_F_8, width = 6, height = 5) 
# ggsave("Results/acute_24hr_F_vol.png", plot = acute_volcano_F_24, width = 6, height = 5) 
# ggsave("Results/acute_72hr_F_vol.png", plot = acute_volcano_F_72, width = 6, height = 5) 
```

#### Males
```{r}
acute <- top_M_8 %>% 
  filter(abs(log2FoldChange) >1.5 & pvalue < 0.05) %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 ~ "Upregulated",
                            log2FoldChange < -1.5 ~ "Downregulated",
                            TRUE ~ "NS"))
acute_genes <- rownames(acute)
color_8 <- "#FEC287"
color_24 <- "#F56B5C"
color_72 <- "#A02F7F"

(acute_volcano_M_8 <- acute %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name), max.overlaps = 2) +
  scale_color_manual(values = c("#000004", "lightgrey", "#FEC287"), name = NULL) +
  theme_bw() +
  xlab("Log2 Fold Change at 8 hours") +
  ylab("-log10(P value)") +
  theme(aspect.ratio = 1) +
  ggtitle("Differentially expressed genes 8 hours \nafter P. rettgeri injection in males"))

colors_time <- c("#A02F7F", "#F56B5C", "#FEC287", "#000004")
acute_M_24 <- top_M_24 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% acute_genes)

(acute_volcano_M_24 <- acute_M_24 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#F56B5C"), name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log2 Fold Change at 24 hours") +
  ylab("-log10(P value)") +
  ggtitle("Acute response genes differentially expressed 24 hours \nafter P. rettgeri injection in males"))

acute_M_72 <- top_M_72 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% acute_genes)

(acute_volcano_M_72 <- acute_M_72 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#A02F7F"), name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log2 Fold Change at 72 hours") +
  ylab("-log10(P value)") +
  ggtitle("Acute response genes differentially expressed 72 hours \nafter P. rettgeri injection in males"))

# ggsave("Results/acute_8hr_M_vol.png", plot = acute_volcano_M_8, width = 6, height = 5)
# ggsave("Results/acute_24hr_M_vol.png", plot = acute_volcano_M_24, width = 6, height = 5)
# ggsave("Results/acute_72hr_M_vol.png", plot = acute_volcano_M_72, width = 6, height = 5)
```


#### GSEA
```{r}
acute <- cbind(acute, Gene_ID_Flybase[match(acute$gene_name, Gene_ID_Flybase$Gene_name), "CG_ID"])
colnames(acute)[ncol(acute)] = "CG_ID"

ranks <- acute$log2FoldChange[!is.na(acute$log2FoldChange)]
names(ranks) <- acute$CG_ID[!is.na(acute$log2FoldChange)]
head(ranks)
#barplot(sort(ranks, decreasing = T))

fgseaRes_acute_8 <- fgsea(t_Drosophila_REACTOME, ranks, minSize = 15, maxSize = 500)
head(fgseaRes_acute_8[order(padj, -abs(NES)), ], n=10)

fgseaRes_KEGG_acute_8 <- fgsea(t_Drosophila_KEGG, ranks, minSize = 15, maxSize = 500)
head(fgseaRes_KEGG_acute_8[order(padj, -abs(NES)), ], n=10)


original_gene_list <- acute$log2FoldChange
names(original_gene_list) <- rownames(acute)
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList = gene_list,
             ont = "ALL",
             keyType = "SYMBOL",
             minGSSize = 3,
             maxGSSize = 800,
             pvalueCutoff = 0.05,
             verbose = TRUE,
             OrgDb = organism,
             pAdjustMethod = "none")

gse_acute_8_hr <- gse

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_acute_8hr_dotplot <- dotplot(gse_acute_8_hr, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched GO terms in acute responses 8 hours post injection"))
#ggsave("Results/limma/gse_acute_8hr_dotplot.png", plot = gse_acute_8hr_dotplot, width = 14, height = 6) 

# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

df2 = acute[acute$gene_name %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_acute_8 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.5,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_kegg_acute_8hr_dotplot <- dotplot(kegg_acute_8, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in acute responses 8 hours post injection"))
#ggsave("Results/limma/gse_kegg_acute_8hr_dotplot.png", plot = gse_kegg_acute_8hr_dotplot, width = 14, height = 6) 
```

##### Female GSEA
```{r}
acute_F <- top_F_8 %>% 
 # filter(pvalue < 0.05) %>% 
  mutate(gene_name = rownames(.))

original_gene_list <- acute_F$log2FoldChange
names(original_gene_list) <- rownames(acute_F)
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList = gene_list,
             ont = "ALL",
             keyType = "SYMBOL",
             minGSSize = 3,
             maxGSSize = 800,
             pvalueCutoff = 0.05,
             verbose = TRUE,
             OrgDb = organism,
             pAdjustMethod = "none")

gse_acute_F <- gse

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_acute_F_dotplot <- dotplot(gse_acute_F, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched GO terms in acute response (female)"))
#ggsave("Results/limma/gse_acute_F_dotplot.png", plot = gse_acute_F_dotplot, width = 14, height = 6) 

# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

df2 = acute_F[acute_F$gene_name %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_acute_F <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_kegg_acute_F_dotplot <- dotplot(kegg_acute_F, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in acute response (females)"))
#ggsave("Results/limma/gse_kegg_acute_F_dotplot.png", plot = gse_kegg_acute_F_dotplot, width = 14, height = 6) 
```

##### Male GSEA
```{r}
acute_M <- top_M_8 %>% 
  #filter(pvalue < 0.05) %>% 
  mutate(gene_name = rownames(.))

original_gene_list <- acute_M$log2FoldChange
names(original_gene_list) <- rownames(acute_M)
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList = gene_list,
             ont = "ALL",
             keyType = "SYMBOL",
             minGSSize = 3,
             maxGSSize = 800,
             pvalueCutoff = 0.05,
             verbose = TRUE,
             OrgDb = organism,
             pAdjustMethod = "none")

gse_acute_M <- gse

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_acute_M_dotplot <- dotplot(gse_acute_M, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched GO terms in acute response (male)"))
#ggsave("Results/limma/gse_acute_M_dotplot.png", plot = gse_acute_M_dotplot, width = 14, height = 6) 

# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

df2 = acute_M[acute_M$gene_name %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_acute_M <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_kegg_acute_M_dotplot <- dotplot(kegg_acute_M, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in acute response (males)"))
#ggsave("Results/limma/gse_kegg_acute_M_dotplot.png", plot = gse_kegg_acute_M_dotplot, width = 14, height = 6) 
```


### Chronic response
#### WGCNA
```{r}
countdata_chronic <- countdata %>% 
  select(contains("72")) %>% 
  select(contains("Y")) %>% 
  select(!contains("WD"))

col_sel <- names(countdata_chronic)

my_data <- countdata_chronic %>% 
  tidyr::pivot_longer(
    .,
    col = all_of(col_sel)
  ) %>% 
  mutate(group = name)

#==== Plot groups (Sample Groups vs RNA Seq Counts) to identify outliers
# (
#  p <- my_data %>%
#     ggplot(., aes(x = name, y = value)) +             # x = treatment, y = RNA Seq count
#     geom_violin() +                                   # violin plot, show distribution
#     geom_point(alpha = 0.2) +                         # scatter plot
#     theme_bw() +
#     theme(
#       axis.text.x = element_text(angle = 90)          # Rotate treatment text
#     ) +
#     labs(x = "Treatment Groups", y = "RNA Seq Counts") +
#     facet_grid(cols = vars(group), drop = TRUE, scales = "free_x")      # Facet by hour
# )
```

```{r}
library(DESeq2)

colData_chronic <- colData %>% 
  filter(Treatment != "WD" & time == 72 & Dissected_Y.N == "Y") %>% 
  mutate(Treatment = factor(Treatment, levels = c("UC", "Pr")),
         condition = factor(paste(Sex, Treatment, sep = "_"), levels = c("F_UC", "M_UC","F_Pr", "M_Pr")))

matrix_countdata_chronic <- as.matrix(countdata_chronic)

dds <- DESeqDataSetFromMatrix(matrix_countdata_chronic,
                              colData_chronic,
                              design = ~condition)
#> converting counts to integer mode
#> Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
#> design formula are characters, converting to factors

dds <- DESeq(dds)

#> estimating size factors
#> estimating dispersions
#> gene-wise dispersion estimates
#> mean-dispersion relationship
#> final dispersion estimates
#> fitting model and testing
vsd <- varianceStabilizingTransformation(dds)
library(genefilter)      # <= why is this here?
#>
#> Attaching package: 'genefilter'
#> The following objects are masked from 'package:matrixStats':
#>
#>     rowSds, rowVars
#> The following object is masked from 'package:readr':
#>
#>     spec
wpn_vsd <- getVarianceStabilizedData(dds)
rv_wpn <- rowVars(wpn_vsd)
summary(rv_wpn)
#>     Min.  1st Qu.   Median     Mean  3rd Qu.     Max.
#>  0.00000  0.00000  0.00000  0.08044  0.03322 11.14529

q75_wpn <- quantile( rowVars(wpn_vsd), .75)  # <= original
q95_wpn <- quantile( rowVars(wpn_vsd), .95)  # <= changed to 95 quantile to reduce dataset
expr_normalized <- wpn_vsd[ rv_wpn > q95_wpn, ]

expr_normalized[1:5,1:10]
#>                       B-3      B-4      B-5      L-3      L-4      L-5      S-3
#> AC149818.2_FG001 7.600901 7.077399 7.803434 7.220840 7.410408 8.028223 7.160846
#> AC149829.2_FG003 8.782014 8.179876 7.900062 8.299778 7.529891 8.631731 8.055118
#> AC182617.3_FG001 8.047244 7.120668 6.885533 7.501391 7.279413 7.809565 7.184253
#> AC186512.3_FG001 6.901539 7.389644 6.975945 6.859593 7.370816 6.633722 7.798843
#> AC186512.3_FG007 7.919688 7.754506 7.670946 7.417760 7.988427 7.904850 7.484542
#>                       S-4      S-5   B_L1.1
#> AC149818.2_FG001 7.401382 7.345322 6.524435
#> AC149829.2_FG003 8.744502 8.142909 8.240407
#> AC182617.3_FG001 8.140134 6.972400 7.777347
#> AC186512.3_FG001 6.949501 6.952659 6.059033
#> AC186512.3_FG007 8.375664 7.762799 6.335663
dim(expr_normalized)
#> [1] 5486   24

expr_normalized_df <- data.frame(expr_normalized) %>%
  mutate(
    Gene_id = row.names(expr_normalized)
  ) %>%
  pivot_longer(-Gene_id)

# expr_normalized_df %>% ggplot(., aes(x = name, y = value)) +
#   geom_violin() +
#   geom_point() +
#   theme_bw() +
#   theme(
#     axis.text.x = element_text( angle = 90)
#   ) +
#   ylim(0, NA) +
#   labs(
#     title = "Normalized and 95 quantile Expression",
#     x = "treatment",
#     y = "normalized expression"
#   )
```


```{r}
input_mat = t(expr_normalized)

input_mat[1:5,1:10]           # Look at first 5 rows and 10 columns

#library(WGCNA)
allowWGCNAThreads()          # allow multi-threading (optional)
#> Allowing multi-threading with up to 4 threads.

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to = 20, by = 2))

# Call the network topology analysis function
sft = pickSoftThreshold(
  input_mat,             # <= Input data
  #blockSize = 30,
  powerVector = powers,
  verbose = 5
  )

par(mfrow = c(1,2));
cex1 = 0.9;

plot(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     xlab = "Soft Threshold (power)",
     ylab = "Scale Free Topology Model Fit, signed R^2",
     main = paste("Scale independence")
)
text(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     labels = powers, cex = cex1, col = "red"
)
abline(h = 0.90, col = "red")
plot(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     xlab = "Soft Threshold (power)",
     ylab = "Mean Connectivity",
     type = "n",
     main = paste("Mean connectivity")
)
text(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     labels = powers,
     cex = cex1, col = "red")
```


```{r}
picked_power = 10
temp_cor <- cor       
cor <- WGCNA::cor         # Force it to use WGCNA cor function (fix a namespace conflict issue)
netwk <- blockwiseModules(input_mat,                # <= input here

                          # == Adjacency Function ==
                          power = picked_power,                # <= power here
                          networkType = "signed",

                          # == Tree and Block Options ==
                          deepSplit = 2,
                          pamRespectsDendro = F,
                          # detectCutHeight = 0.75,
                          minModuleSize = 30,
                          maxBlockSize = 4000,

                          # == Module Adjustments ==
                          reassignThreshold = 0,
                          mergeCutHeight = 0.25,

                          # == TOM == Archive the run results in TOM file (saves time)
                          saveTOMs = T,
                          saveTOMFileBase = "ER",

                          # == Output Options
                          numericLabels = T,
                          verbose = 3)


cor <- temp_cor     # Return cor function to original namespace
```


```{r}
# Convert labels to colors for plotting
mergedColors = labels2colors(netwk$colors)
# Plot the dendrogram and the module colors underneath
plotDendroAndColors(
  netwk$dendrograms[[1]],
  mergedColors[netwk$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05 )
```

```{r}
chronic_module_df_95 <- data.frame(
  gene_name = names(netwk$colors),
  colors = labels2colors(netwk$colors)
)

chronic_module_df_95[1:5,]
#>            gene_id    colors
#> 1 AC149818.2_FG001      blue
#> 2 AC149829.2_FG003      blue
#> 3 AC182617.3_FG001      blue
#> 4 AC186512.3_FG001 turquoise
#> 5 AC186512.3_FG007 turquoise

# write_delim(module_df,
#            file = "Results/gene_modules.txt",
#            delim = "\t")
```


```{r}
# Get Module Eigengenes per cluster
MEs0 <- moduleEigengenes(input_mat, mergedColors)$eigengenes

# Reorder modules so similar modules are next to each other
MEs0 <- orderMEs(MEs0)
module_order = names(MEs0) %>% gsub("ME","", .)

# Add treatment names
MEs0$treatment = row.names(MEs0)

modules_of_interest = c("brown", "blue", "turquoise")

# tidy & plot data
mME = MEs0 %>%
  pivot_longer(-treatment) %>%
  mutate(
    name = gsub("ME", "", name),
    name = factor(name, levels = module_order)
  ) %>% 
  filter(name %in% modules_of_interest)

(module_heatmap <- mME %>% ggplot(., aes(x=treatment, y=name, fill=value)) +
  geom_tile() +
  theme_bw() +
  scale_fill_viridis() +
  theme(axis.text.x = element_text(angle=90)) +
  labs(title = "Module-trait Relationships", y = "Modules", fill="corr"))
#ggsave(filename = "Results/chronic/module_heatmap.png", module_heatmap, width = 12, height = 4)
```


```{r}
# pick out a few modules of interest here
modules_of_interest = c("brown", "blue", "turquoise")

# Pull out list of genes in that module
submod = chronic_module_df_95 %>%
  subset(colors %in% modules_of_interest)

row.names(chronic_module_df_95) = chronic_module_df_95$gene_name

# Get normalized expression for those genes
expr_normalized[1:5,1:10]

subexpr = expr_normalized[submod$gene_name,]

submod_df = data.frame(subexpr) %>%
  mutate(
    gene_name = row.names(.)
  ) %>%
  pivot_longer(-gene_name) %>%
  mutate(
    module = chronic_module_df_95[gene_name,]$colors,
    module = factor(module, levels = rev(module_order))
  )

(expression_plot <- submod_df %>% ggplot(., aes(x=name, y=value, group=gene_name)) +
  geom_line(aes(color = module),
            alpha = 0.2) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90)
  ) +
  facet_grid(rows = vars(module), scales = "free") +
  scale_color_manual(values = c("turquoise", "brown", "blue")) +
  labs(x = "treatment",
       y = "normalized expression"))
#ggsave("Results/chronic/expression_module.png", expression_plot, width = 8, height = 6)
```

```{r}
genes_of_interest = chronic_module_df_95 %>%
  subset(colors %in% modules_of_interest)

expr_of_interest = expr_normalized[genes_of_interest$gene_name,]
expr_of_interest[1:5,1:5]


# Only recalculate TOM for modules of interest (faster, altho there's some online discussion if this will be slightly off)
TOM = TOMsimilarityFromExpr(t(expr_of_interest),
                            power = picked_power)
#> TOM calculation: adjacency..
#> ..will use 4 parallel threads.
#>  Fraction of slow calculations: 0.000000
#> ..connectivity..
#> ..matrix multiplication (system BLAS)..
#> ..normalization..
#> ..done.

# Add gene names to row and columns
row.names(TOM) = row.names(expr_of_interest)
colnames(TOM) = row.names(expr_of_interest)

edge_list = data.frame(TOM) %>%
  mutate(
    gene1 = row.names(.)
  ) %>%
  pivot_longer(-gene1) %>%
  dplyr::rename(gene2 = name, correlation = value) %>%
  unique() %>%
  subset(!(gene1==gene2)) %>%
  mutate(
    module1 = chronic_module_df_95[gene1,]$colors,
    module2 = chronic_module_df_95[gene2,]$colors
  )

head(edge_list)

 # write_delim(edge_list,
 #              file = "Results/chronic/edgelist.csv",
 #              delim = ",")
```

##### Characterizing modules
```{r}
chronic_wgcna_kegg <- vector("list")


for(i in modules_of_interest){
  genes_df <- chronic_module_df_95 %>% 
    filter(colors == i)
  original_gene_list <- genes_df$colors
names(original_gene_list) <- genes_df$gene_name

ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

gene <- dedup_ids$ENTREZID

kk <- enrichKEGG(gene = gene,
                 organism = "dme",
                 keyType = "ncbi-geneid",
                 pvalueCutoff = 0.05)
chronic_wgcna_kegg[[i]] <- kk


}

(kegg_chronic_brown <- dotplot(chronic_wgcna_kegg$brown, showCategory=10) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in brown module"))
#ggsave("Results/chronic/kegg_chronic_brown.png", plot = kegg_chronic_brown, width = 8, height = 6) 
(kegg_chronic_blue <- dotplot(chronic_wgcna_kegg$blue, showCategory=10) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in blue module"))
#ggsave("Results/chronic/kegg_chronic_blue.png", plot = kegg_chronic_blue, width = 8, height = 6) 
(kegg_chronic_turquoise <- dotplot(chronic_wgcna_kegg$turquoise, showCategory=10) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in turquoise module"))
#ggsave("Results/chronic/kegg_chronic_turquoise.png", plot = kegg_chronic_turquoise, width = 8, height = 6) 


```




```{r}
chronic <- top_72 %>% 
  filter(pvalue < 0.05) %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 ~ "Upregulated",
                            log2FoldChange < -1.5 ~ "Downregulated",
                            TRUE ~ "NS"))
chronic_genes <- rownames(chronic)

(chronic_volcano_72 <- chronic %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#A02F7F"), name = NULL) +
  theme_bw() +
  xlab("Log2 Fold Change at 72 hours") +
  ylab("-log10(P value)") +
  ggtitle("Chronic response genes differentially expressed 72 hours after P. rettgeri injection"))

colors_time <- c("#A02F7F", "#F56B5C", "#FEC287", "#000004")
chronic_24 <- top_24 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% chronic_genes)

(chronic_volcano_24 <- chronic_24 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#F56B5C"), name = NULL) +
  theme_bw() +
  xlab("Log2 Fold Change at 24 hours") +
  ylab("-log10(P value)") +
  ggtitle("Chronic response genes differentially expressed 24 hours after P. rettgeri injection"))

chronic_8 <- top_8 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% chronic_genes)

(chronic_volcano_8 <- chronic_8 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#FEC287"), name = NULL) +
  theme_bw() +
  xlab("Log2 Fold Change at 8 hours") +
  ylab("-log10(P value)") +
  ggtitle("Chronic response genes differentially expressed 8 hours after P. rettgeri injection"))

 # ggsave("Results/chronic_8.png", chronic_volcano_8, width = 8, height = 6)
 # ggsave("Results/chronic_24.png", chronic_volcano_24, width = 8, height = 6)
 # ggsave("Results/chronic_72.png", chronic_volcano_72, width = 8, height = 6)
```


#### Female
```{r}

chronic_F <- top_F_72 %>% 
  filter(pvalue < 0.05) %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 ~ "Upregulated",
                            log2FoldChange < -1.5 ~ "Downregulated",
                            TRUE ~ "NS"))

chronic_genes <- rownames(chronic_F)

(chronic_F_volcano_72 <- chronic_F %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#A02F7F"), name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log2 Fold Change at 72 hours") +
  ylab("-log10(P value)") +
  ggtitle("Chronic response genes differentially expressed 72 hours \nafter P. rettgeri injection (females)"))

#colors_time <- c("#A02F7F", "#F56B5C", "#FEC287", "#000004")
chronic_F_24 <- top_F_24 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% chronic_genes)

(chronic_F_volcano_24 <- chronic_F_24 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#F56B5C"), name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log2 Fold Change at 24 hours") +
  ylab("-log10(P value)") +
  ggtitle("Chronic response genes differentially expressed 24 hours \nafter P. rettgeri injection (females)"))

chronic_F_8 <- top_F_8 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% chronic_genes)

(chronic_F_volcano_8 <- chronic_F_8 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#FEC287"), name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log2 Fold Change at 8 hours") +
  ylab("-log10(P value)") +
  ggtitle("Chronic response genes differentially expressed 8 hours \nafter P. rettgeri injection (females)"))

 ggsave("Results/chronic_F_8.png", chronic_F_volcano_8, width = 8, height = 6)
 ggsave("Results/chronic_F_24.png", chronic_F_volcano_24, width = 8, height = 6)
 ggsave("Results/chronic_F_72.png", chronic_F_volcano_72, width = 8, height = 6)
```

#### Male
```{r}

chronic_M <- top_M_72 %>% 
  filter(pvalue < 0.05) %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 ~ "Upregulated",
                            log2FoldChange < -1.5 ~ "Downregulated",
                            TRUE ~ "NS"))

chronic_genes <- rownames(chronic_M)

(chronic_M_volcano_72 <- chronic_M %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#A02F7F"), name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log2 Fold Change at 72 hours") +
  ylab("-log10(P value)") +
  ggtitle("Chronic response genes differentially expressed 72 hours \nafter P. rettgeri injection (males)"))

#colors_time <- c("#A02F7F", "#F56B5C", "#FEC287", "#000004")
chronic_M_24 <- top_M_24 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% chronic_genes)

(chronic_M_volcano_24 <- chronic_M_24 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#F56B5C"), name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log2 Fold Change at 24 hours") +
  ylab("-log10(P value)") +
  ggtitle("Chronic response genes differentially expressed 24 hours \nafter P. rettgeri injection (males)"))

chronic_M_8 <- top_M_8 %>% 
  mutate(gene_name = rownames(.),
         colors = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                            log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated", 
                            TRUE ~ "NS")) %>% 
  filter(gene_name %in% chronic_genes)

(chronic_M_volcano_8 <- chronic_M_8 %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = colors)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#FEC287"), name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log2 Fold Change at 8 hours") +
  ylab("-log10(P value)") +
  ggtitle("Chronic response genes differentially expressed 8 hours \nafter P. rettgeri injection (males)"))
# 
#  ggsave("Results/chronic_M_8.png", chronic_M_volcano_8, width = 8, height = 6)
#  ggsave("Results/chronic_M_24.png", chronic_M_volcano_24, width = 8, height = 6)
#  ggsave("Results/chronic_M_72.png", chronic_M_volcano_72, width = 8, height = 6)
```



#### GSEA
```{r}
chronic <- cbind(chronic, Gene_ID_Flybase[match(chronic$gene_name, Gene_ID_Flybase$Gene_name), "CG_ID"])
colnames(chronic)[ncol(chronic)] = "CG_ID"

ranks <- chronic$log2FoldChange[!is.na(chronic$log2FoldChange)]
names(ranks) <- chronic$CG_ID[!is.na(chronic$log2FoldChange)]
head(ranks)
#barplot(sort(ranks, decreasing = T))

fgseaRes_chronic_8 <- fgsea(t_Drosophila_REACTOME, ranks, minSize = 3, maxSize = 500)
head(fgseaRes_chronic_8[order(padj, -abs(NES)), ], n=10)

fgseaRes_KEGG_chronic_8 <- fgsea(t_Drosophila_KEGG, ranks, minSize = 3, maxSize = 500)
head(fgseaRes_KEGG_chronic_8[order(padj, -abs(NES)), ], n=10)


original_gene_list <- chronic$log2FoldChange
names(original_gene_list) <- rownames(chronic)
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList = gene_list,
             ont = "ALL",
             keyType = "SYMBOL",
             minGSSize = 3,
             maxGSSize = 800,
             pvalueCutoff = 0.05,
             verbose = TRUE,
             OrgDb = organism,
             pAdjustMethod = "none")

gse_chronic_72_hr <- gse

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_chronic_72hr_dotplot <- dotplot(gse_chronic_72_hr, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched GO terms in chronic responses 72 hours post injection"))
#ggsave("Results/limma/gse_chronic_72hr_dotplot.png", plot = gse_chronic_72hr_dotplot, width = 14, height = 6) 

# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

df2 = chronic[chronic$gene_name %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_chronic_72 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.5,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_kegg_chronic_72hr_dotplot <- dotplot(kegg_chronic_72, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in chronic responses 72 hours post injection"))
#ggsave("Results/limma/gse_kegg_chronic_72hr_dotplot.png", plot = gse_kegg_chronic_72hr_dotplot, width = 14, height = 6) 
```

```{r}
original_gene_list <- top_72$log2FoldChange
names(original_gene_list) <- rownames(top_72)
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList = gene_list,
             ont = "ALL",
             keyType = "SYMBOL",
             minGSSize = 3,
             maxGSSize = 800,
             pvalueCutoff = 0.05,
             verbose = TRUE,
             OrgDb = organism,
             pAdjustMethod = "none")

gse_72_hr <- gse

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_72hr_dotplot <- dotplot(gse_72_hr, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched GO terms 72 hours post injection"))
#ggsave("Results/limma/gse_72hr_dotplot.png", plot = gse_72hr_dotplot, width = 14, height = 6) 

# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]
top_72$gene_name <- rownames(top_72)

df2 = top_72[top_72$gene_name %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_72 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_kegg_72hr_dotplot <- dotplot(kegg_72, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways 72 hours post injection"))
#ggsave("Results/limma/gse_kegg_72hr_dotplot.png", plot = gse_kegg_72hr_dotplot, width = 14, height = 6) 

```

```{r}
downreg_chronic <- top_72 %>% 
    filter(pvalue < 0.05 & log2FoldChange <0)
  original_gene_list <- downreg_chronic$log2FoldChange
names(original_gene_list) <- downreg_chronic$gene_name

ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

gene <- dedup_ids$ENTREZID

kk <- enrichKEGG(gene = gene,
                 organism = "dme",
                 keyType = "ncbi-geneid",
                 pvalueCutoff = 0.05)
downreg_chronic_kegg <- kk

(downreg_chronic_vol <- ggplot(downreg_chronic, aes(x = log2FoldChange, y = -log(pvalue))) +
  geom_point(color = color_72) +
  geom_text_repel(aes(label = gene_name)) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Significantly downregulated genes during the chronic stage"))
#ggsave("Results/downreg_chronic_vol.png", plot = downreg_chronic_vol, width = 8, height = 6) 

```




#####Female GSEA
```{r}
original_gene_list <- top_F_72$log2FoldChange
names(original_gene_list) <- rownames(top_F_72)
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList = gene_list,
             ont = "ALL",
             keyType = "SYMBOL",
             minGSSize = 3,
             maxGSSize = 800,
             pvalueCutoff = 0.05,
             verbose = TRUE,
             OrgDb = organism,
             pAdjustMethod = "none")

gse_F_72_hr <- gse

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_F_72hr_dotplot <- dotplot(gse_F_72_hr, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched GO terms 72 hours post injection (females)"))
#ggsave("Results/limma/gse_F_72hr_dotplot.png", plot = gse_F_72hr_dotplot, width = 14, height = 6) 

# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]
top_F_72$gene_name <- rownames(top_F_72)

df2 = top_F_72[top_F_72$gene_name %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_F_72 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_kegg_F_72hr_dotplot <- dotplot(kegg_F_72, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways 72 hours post injection (females)"))
#ggsave("Results/limma/gse_kegg_F_72hr_dotplot.png", plot = gse_kegg_F_72hr_dotplot, width = 14, height = 6) 

```

 
##### Male GSEA
```{r}
original_gene_list <- top_M_72$log2FoldChange
names(original_gene_list) <- rownames(top_M_72)
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList = gene_list,
             ont = "ALL",
             keyType = "SYMBOL",
             minGSSize = 3,
             maxGSSize = 800,
             pvalueCutoff = 0.05,
             verbose = TRUE,
             OrgDb = organism,
             pAdjustMethod = "none")

gse_M_72_hr <- gse

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_M_72hr_dotplot <- dotplot(gse_M_72_hr, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched GO terms 72 hours post injection (males)"))
#ggsave("Results/limma/gse_M_72hr_dotplot.png", plot = gse_M_72hr_dotplot, width = 14, height = 6) 

# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]
top_M_72$gene_name <- rownames(top_M_72)

df2 = top_M_72[top_M_72$gene_name %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_M_72 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(gse_kegg_M_72hr_dotplot <- dotplot(kegg_M_72, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways 72 hours post injection"))
#ggsave("Results/limma/gse_kegg_M_72hr_dotplot.png", plot = gse_kegg_M_72hr_dotplot, width = 14, height = 6) 

```

### Middle response
#### WGCNA
```{r}
countdata_mid <- countdata %>% 
  select(contains("24")) %>% 
  select(contains("Y")) %>% 
  select(!contains("WD"))

col_sel <- names(countdata_mid)

my_data <- countdata_mid %>% 
  tidyr::pivot_longer(
    .,
    col = all_of(col_sel)
  ) %>% 
  mutate(group = name)

#==== Plot groups (Sample Groups vs RNA Seq Counts) to identify outliers
# (
#  p <- my_data %>%
#     ggplot(., aes(x = name, y = value)) +             # x = treatment, y = RNA Seq count
#     geom_violin() +                                   # violin plot, show distribution
#     geom_point(alpha = 0.2) +                         # scatter plot
#     theme_bw() +
#     theme(
#       axis.text.x = element_text(angle = 90)          # Rotate treatment text
#     ) +
#     labs(x = "Treatment Groups", y = "RNA Seq Counts") +
#     facet_grid(cols = vars(group), drop = TRUE, scales = "free_x")      # Facet by hour
# )
```

```{r}
library(DESeq2)

colData_mid <- colData %>% 
  filter(Treatment != "WD" & time == 24 & Dissected_Y.N == "Y") %>% 
  mutate(Treatment = factor(Treatment, levels = c("UC", "Pr")),
         condition = factor(paste(Sex, Treatment, sep = "_"), levels = c("F_UC", "M_UC","F_Pr", "M_Pr")))

matrix_countdata_mid <- as.matrix(countdata_mid)

dds <- DESeqDataSetFromMatrix(matrix_countdata_mid,
                              colData_mid,
                              design = ~condition)
#> converting counts to integer mode
#> Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
#> design formula are characters, converting to factors

dds <- DESeq(dds)

#> estimating size factors
#> estimating dispersions
#> gene-wise dispersion estimates
#> mean-dispersion relationship
#> final dispersion estimates
#> fitting model and testing
vsd <- varianceStabilizingTransformation(dds)
library(genefilter)      # <= why is this here?
#>
#> Attaching package: 'genefilter'
#> The following objects are masked from 'package:matrixStats':
#>
#>     rowSds, rowVars
#> The following object is masked from 'package:readr':
#>
#>     spec
wpn_vsd <- getVarianceStabilizedData(dds)
rv_wpn <- rowVars(wpn_vsd)
summary(rv_wpn)
#>     Min.  1st Qu.   Median     Mean  3rd Qu.     Max.
#>  0.00000  0.00000  0.00000  0.08044  0.03322 11.14529

q75_wpn <- quantile( rowVars(wpn_vsd), .75)  # <= original
q95_wpn <- quantile( rowVars(wpn_vsd), .95)  # <= changed to 95 quantile to reduce dataset
expr_normalized <- wpn_vsd[ rv_wpn > q95_wpn, ]

expr_normalized[1:5,1:10]
#>                       B-3      B-4      B-5      L-3      L-4      L-5      S-3
#> AC149818.2_FG001 7.600901 7.077399 7.803434 7.220840 7.410408 8.028223 7.160846
#> AC149829.2_FG003 8.782014 8.179876 7.900062 8.299778 7.529891 8.631731 8.055118
#> AC182617.3_FG001 8.047244 7.120668 6.885533 7.501391 7.279413 7.809565 7.184253
#> AC186512.3_FG001 6.901539 7.389644 6.975945 6.859593 7.370816 6.633722 7.798843
#> AC186512.3_FG007 7.919688 7.754506 7.670946 7.417760 7.988427 7.904850 7.484542
#>                       S-4      S-5   B_L1.1
#> AC149818.2_FG001 7.401382 7.345322 6.524435
#> AC149829.2_FG003 8.744502 8.142909 8.240407
#> AC182617.3_FG001 8.140134 6.972400 7.777347
#> AC186512.3_FG001 6.949501 6.952659 6.059033
#> AC186512.3_FG007 8.375664 7.762799 6.335663
dim(expr_normalized)
#> [1] 5486   24

expr_normalized_df <- data.frame(expr_normalized) %>%
  mutate(
    Gene_id = row.names(expr_normalized)
  ) %>%
  pivot_longer(-Gene_id)

# expr_normalized_df %>% ggplot(., aes(x = name, y = value)) +
#   geom_violin() +
#   geom_point() +
#   theme_bw() +
#   theme(
#     axis.text.x = element_text( angle = 90)
#   ) +
#   ylim(0, NA) +
#   labs(
#     title = "Normalized and 95 quantile Expression",
#     x = "treatment",
#     y = "normalized expression"
#   )
```


```{r}
input_mat = t(expr_normalized)

input_mat[1:5,1:10]           # Look at first 5 rows and 10 columns

#library(WGCNA)
allowWGCNAThreads()          # allow multi-threading (optional)
#> Allowing multi-threading with up to 4 threads.

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to = 20, by = 2))

# Call the network topology analysis function
sft = pickSoftThreshold(
  input_mat,             # <= Input data
  #blockSize = 30,
  powerVector = powers,
  verbose = 5
  )

par(mfrow = c(1,2));
cex1 = 0.9;

plot(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     xlab = "Soft Threshold (power)",
     ylab = "Scale Free Topology Model Fit, signed R^2",
     main = paste("Scale independence")
)
text(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     labels = powers, cex = cex1, col = "red"
)
abline(h = 0.90, col = "red")
plot(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     xlab = "Soft Threshold (power)",
     ylab = "Mean Connectivity",
     type = "n",
     main = paste("Mean connectivity")
)
text(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     labels = powers,
     cex = cex1, col = "red")
```


```{r}
picked_power = 16
temp_cor <- cor       
cor <- WGCNA::cor         # Force it to use WGCNA cor function (fix a namespace conflict issue)
netwk <- blockwiseModules(input_mat,                # <= input here

                          # == Adjacency Function ==
                          power = picked_power,                # <= power here
                          networkType = "signed",

                          # == Tree and Block Options ==
                          deepSplit = 2,
                          pamRespectsDendro = F,
                          # detectCutHeight = 0.75,
                          minModuleSize = 30,
                          maxBlockSize = 4000,

                          # == Module Adjustments ==
                          reassignThreshold = 0,
                          mergeCutHeight = 0.25,

                          # == TOM == Archive the run results in TOM file (saves time)
                          saveTOMs = T,
                          saveTOMFileBase = "ER",

                          # == Output Options
                          numericLabels = T,
                          verbose = 3)


cor <- temp_cor     # Return cor function to original namespace
```


```{r}
# Convert labels to colors for plotting
mergedColors = labels2colors(netwk$colors)
# Plot the dendrogram and the module colors underneath
plotDendroAndColors(
  netwk$dendrograms[[1]],
  mergedColors[netwk$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05 )
```

```{r}
mid_module_df_95 <- data.frame(
  gene_name = names(netwk$colors),
  colors = labels2colors(netwk$colors)
)

mid_module_df_95[1:5,]
#>            gene_id    colors
#> 1 AC149818.2_FG001      blue
#> 2 AC149829.2_FG003      blue
#> 3 AC182617.3_FG001      blue
#> 4 AC186512.3_FG001 turquoise
#> 5 AC186512.3_FG007 turquoise

# write_delim(module_df,
#            file = "Results/gene_modules.txt",
#            delim = "\t")
```


```{r}
# Get Module Eigengenes per cluster
MEs0 <- moduleEigengenes(input_mat, mergedColors)$eigengenes

# Reorder modules so similar modules are next to each other
MEs0 <- orderMEs(MEs0)
module_order = names(MEs0) %>% gsub("ME","", .)

# Add treatment names
MEs0$treatment = row.names(MEs0)

modules_of_interest = c("grey", "turquoise", "blue", "yellow")

# tidy & plot data
mME = MEs0 %>%
  pivot_longer(-treatment) %>%
  mutate(
    name = gsub("ME", "", name),
    name = factor(name, levels = module_order)
  ) %>% 
  filter(name %in% modules_of_interest)

(module_heatmap <- mME %>% ggplot(., aes(x=treatment, y=name, fill=value)) +
  geom_tile() +
  theme_bw() +
  scale_fill_viridis() +
  theme(axis.text.x = element_text(angle=90)) +
  labs(title = "Module-trait Relationships", y = "Modules", fill="corr"))
#ggsave(filename = "Results/mid/module_heatmap.png", module_heatmap, width = 12, height = 4)
```


```{r}
# pick out a few modules of interest here
modules_of_interest = c("grey", "turquoise", "blue", "yellow")

# Pull out list of genes in that module
submod = mid_module_df_95 %>%
  subset(colors %in% modules_of_interest)

row.names(mid_module_df_95) = mid_module_df_95$gene_name

# Get normalized expression for those genes
expr_normalized[1:5,1:10]

subexpr = expr_normalized[submod$gene_name,]

submod_df = data.frame(subexpr) %>%
  mutate(
    gene_name = row.names(.)
  ) %>%
  pivot_longer(-gene_name) %>%
  mutate(
    module = mid_module_df_95[gene_name,]$colors,
    module = factor(module, levels = rev(module_order))
  )

(expression_plot <- submod_df %>% ggplot(., aes(x=name, y=value, group=gene_name)) +
  geom_line(aes(color = module),
            alpha = 0.2) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90)
  ) +
  facet_grid(rows = vars(module), scales = "free") +
  scale_color_manual(values = modules_of_interest) +
  labs(x = "treatment",
       y = "normalized expression"))
#ggsave("Results/mid/expression_module.png", expression_plot, width = 8, height = 6)
```

```{r}
genes_of_interest = mid_module_df_95 %>%
  subset(colors %in% modules_of_interest)

expr_of_interest = expr_normalized[genes_of_interest$gene_name,]
expr_of_interest[1:5,1:5]


# Only recalculate TOM for modules of interest (faster, altho there's some online discussion if this will be slightly off)
TOM = TOMsimilarityFromExpr(t(expr_of_interest),
                            power = picked_power)
#> TOM calculation: adjacency..
#> ..will use 4 parallel threads.
#>  Fraction of slow calculations: 0.000000
#> ..connectivity..
#> ..matrix multiplication (system BLAS)..
#> ..normalization..
#> ..done.

# Add gene names to row and columns
row.names(TOM) = row.names(expr_of_interest)
colnames(TOM) = row.names(expr_of_interest)

edge_list = data.frame(TOM) %>%
  mutate(
    gene1 = row.names(.)
  ) %>%
  pivot_longer(-gene1) %>%
  dplyr::rename(gene2 = name, correlation = value) %>%
  unique() %>%
  subset(!(gene1==gene2)) %>%
  mutate(
    module1 = mid_module_df_95[gene1,]$colors,
    module2 = mid_module_df_95[gene2,]$colors
  )

head(edge_list)

  # write_delim(edge_list,
  #              file = "Results/mid/edgelist.csv",
  #              delim = ",")
```

##### Characterizing modules
```{r}
mid_wgcna_kegg <- vector("list")


for(i in modules_of_interest){
  genes_df <- mid_module_df_95 %>% 
    filter(colors == i)
  original_gene_list <- genes_df$colors
names(original_gene_list) <- genes_df$gene_name

ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

gene <- dedup_ids$ENTREZID

kk <- enrichKEGG(gene = gene,
                 organism = "dme",
                 keyType = "ncbi-geneid",
                 pvalueCutoff = 0.05)
mid_wgcna_kegg[[i]] <- kk


}

(kegg_mid_grey <- dotplot(mid_wgcna_kegg$grey, showCategory=10) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in grey module"))
#ggsave("Results/mid/kegg_mid_grey.png", plot = kegg_mid_grey, width = 8, height = 6) 
(kegg_mid_blue <- dotplot(mid_wgcna_kegg$blue, showCategory=10) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in blue module"))
#ggsave("Results/mid/kegg_mid_blue.png", plot = kegg_mid_blue, width = 8, height = 6) 
(kegg_mid_turquoise <- dotplot(mid_wgcna_kegg$turquoise, showCategory=10) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in turquoise module"))
#ggsave("Results/mid/kegg_mid_turquoise.png", plot = kegg_mid_turquoise, width = 8, height = 6) 
(kegg_mid_yellow <- dotplot(mid_wgcna_kegg$yellow, showCategory=10) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched KEGG pathways in yellow module"))
#ggsave("Results/mid/kegg_mid_yellow.png", plot = kegg_mid_yellow, width = 8, height = 6) 


```


 
## Toll and Imd
```{r}
toll_imd <- kegg_8@geneSets$dme04624
toll_imd_genes <- bitr(toll_imd, fromType = "ENTREZID", toType = "SYMBOL", OrgDb=organism)

data_colnames <- colnames(countdata_pr)

gg_toll <- countdata_pr %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(data_colnames), names_to = "Sample", values_to = "Count") %>% 
  #left_join(top_time) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  filter(gene_name %in% toll_imd_genes$SYMBOL) %>% 
  mutate(Challenge = factor(Challenge, levels = c("UC", "Pr")))

(toll_imd_plot <- ggplot(gg_toll, aes(x = Challenge, y = Count)) +
  geom_line() +
  geom_point(aes(y = Count)) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_wrap(~Time, scales = "free") +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection"))
#ggsave(filename = "Results/toll_imd_raw.png", toll_imd_plot, width= 16, height = 12)

# Libraries
library(GGally)

parallel_imd_toll <- gg_toll %>% 
  dplyr::select(-avg_count) %>% 
  pivot_wider(names_from = Challenge, values_from = Count) %>% 
  mutate(Time_factor = factor(Time),
         Sex = factor(Sex))

colors_time <- c("#FEC287", "#F56B5C", "#A02F7F")
sex_labs <- c("Female", "Male")
names(sex_labs) <- c("1", "2")
time_labs <- c("8 hours", "24 hours", "72 hours")
names(time_labs) <- c("8", "24", "72")
# Plot
(parallel_toll_imd_plot <- ggparcoord(parallel_imd_toll,
    columns = 6:7, groupColumn = 8, 
    scale = "globalminmax",
    showPoints = TRUE,
    alphaLines = 0.3
    ) +
    geom_text_repel(aes(label = gene_name)) +
  facet_wrap(Sex~Time, labeller = labeller(Sex = sex_labs, Time = time_labs)) +
  theme_bw() +
  ylab("Gene counts") +
  xlab("Challenge (Unchallenged vs P. rettgeri)") +
  theme(aspect.ratio = 1, legend.position = "none", strip.background = element_rect(fill = "white")) +
  scale_color_manual(values = colors_time) +
  ggtitle("Expression of genes in Toll and Imd pathways at 8, 24, and 72 hours\n after P. rettgeri injection"))
#ggsave(filename = "Results/parallel_toll_imd_plot.png", parallel_toll_imd_plot, width= 8, height = 8)



matrix_countdata_pr <- as.matrix(countdata_pr)

matrix_countdata_pr <- matrix_countdata_pr - rowMeans(matrix_countdata_pr)

gg_toll <- as.data.frame(matrix_countdata_pr) %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(data_colnames), names_to = "Sample", values_to = "Count") %>% 
  #left_join(top_time) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
 # mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  filter(gene_name %in% toll_imd_genes$SYMBOL) %>% 
  mutate(Challenge = factor(Challenge, levels = c("UC", "Pr")),
         Time = factor(Time, levels = c("8", "24", "72")))


colors_time <- c("#FEC287", "#F56B5C", "#A02F7F")
sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")
time_labs <- c("8 hours", "24 hours", "72 hours")
names(time_labs) <- c("8", "24", "72")
(parallel_toll_imd_plot <- ggplot(gg_toll, aes(x = Challenge, y = Count)) +
  geom_line(aes(group = gene_name, y = avg_count, color = Time)) +
  geom_point(aes(color = Time)) +
  geom_text_repel(aes(label = gene_name)) +
  facet_grid(Sex~Time, labeller = labeller(Sex = sex_labs, Time = time_labs)) +
  scale_color_manual(values = colors_time) +
  theme_bw() +
  ylab("Normalized gene counts") +
  theme(aspect.ratio = 1, legend.position = "none", strip.background = element_rect(fill = "white")))
#ggsave(filename = "Results/parallel_toll_imd_plot.png", parallel_toll_imd_plot, width= 8, height = 6)

toll_8_72_vol <- top_time %>% 
  filter(gene_name %in% toll_imd_genes$SYMBOL) %>% 
  mutate(direction = case_when(Pr_72.Pr_8 > 2 & pvalue < 0.05 ~ "Higher during chronic",
                               Pr_72.Pr_8 < -2 & pvalue < 0.05 ~ "Higher during acute",
                               TRUE ~ "NS"))

(toll_8_72_vol_plot <- ggplot(toll_8_72_vol, aes(x = Pr_72.Pr_8, y = -log(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(aes(label = gene_name)) +
  scale_color_manual(values = c("#FEC287", "#A02F7F", "lightgrey"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("Log fold change between infection at 72 hours and 8 hours") +
  ggtitle("Difference in Toll and Imd pathways between \nacute and chronic infection states"))
#ggsave(filename = "Results/toll_8_72_vol_plot.png", toll_8_72_vol_plot, width= 8, height = 6)

acute <- top_8
colnames(acute) <- paste(colnames(acute),"acute",sep="_")
acute$gene_name <- rownames(acute)
chronic <- top_72 %>% 
  select(-gene_name)
colnames(chronic) <- paste(colnames(chronic),"chronic",sep="_")
chronic$gene_name <- rownames(chronic)


acute_v_chronic <- acute %>% 
   left_join(chronic, by='gene_name') %>% 
   mutate(direction_8 = case_when(log2FoldChange_acute > 1.5 & pvalue_acute < 0.05 ~ "Upregulated (acute)",
                                  log2FoldChange_acute < -1.5 & pvalue_acute < 0.05 ~ "Downregulated (acute)",
                                  TRUE ~ "NS"),
          direction_72 = case_when(log2FoldChange_chronic > 1.5 & pvalue_chronic < 0.05 ~ "Upregulated (chronic)",
                                  log2FoldChange_chronic < -1.5 & pvalue_chronic < 0.05 ~ "Downregulated (chronic)",
                                  TRUE ~ "NS"),
          direction = case_when(log2FoldChange_acute > log2FoldChange_chronic ~ "Higher in acute",
                                log2FoldChange_chronic > log2FoldChange_acute ~ "Higher in chronic")) %>% 
  filter(pvalue_acute < 0.05 | pvalue_chronic < 0.05)


(acute_v_chronic_plot <- ggplot(acute_v_chronic, aes(x = log2FoldChange_acute, y = log2FoldChange_chronic)) +
  geom_point(aes(color = direction)) +
  geom_abline(color = "lightgrey") +
  geom_text_repel(aes(label = gene_name)) +
  xlab("Log fold change during acute infection") +
  ylab("Log fold change during chronic infection") +
  scale_color_manual(values = c("#FEC287", "#A02F7F", "lightgrey"), name = "") +
  ylim(c(0, max(acute_v_chronic$log2FoldChange_acute))) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Difference in gene induction between acute and chronic stages"))
#ggsave(filename = "Results/acute_v_chronic_plot.png", acute_v_chronic_plot, width= 8, height = 6)


(acute_v_chronic <- ggplot((data = top_time %>% filter(pvalue < 0.05) %>% mutate(direction = case_when(Pr_72.Pr_8 > 2 & pvalue < 0.05 ~ "Higher during chronic",
                               Pr_72.Pr_8 < -2 & pvalue < 0.05 ~ "Higher during acute",
                               TRUE ~ "NS"))), aes(x = Pr_8.UC_8, y = Pr_72.UC_72)) +
  geom_point(aes(color = direction)) +
  geom_abline(color = "lightgrey") +
  geom_text_repel(aes(label = gene_name)) +
  xlab("Log fold change during acute infection") +
  ylab("Log fold change during chronic infection") +
  scale_color_manual(values = c("#FEC287", "#A02F7F", "lightgrey"), name = "") +
  ylim(c(0, max(top_time$Pr_8.UC_8))) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Difference in gene induction \nbetween acute and chronic stages"))
#ggsave(filename = "Results/acute_v_chronic.png", acute_v_chronic, width= 8, height = 6)

sig_toll_8 <- top_8 %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(gene_name %in% toll_imd_genes$SYMBOL & pvalue < 0.05)

sig_toll_72 <- top_72 %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(gene_name %in% toll_imd_genes$SYMBOL & pvalue < 0.05)

toll_imd_8 <- top_8 %>% 
  mutate(gene_name = rownames(.),
         direction = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                          log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated",
                          TRUE ~ "NS")) %>% 
  filter(gene_name %in% toll_imd_genes$SYMBOL) 

(toll_imd_8_plot <- ggplot(toll_imd_8, aes(x = log2FoldChange, y = -log(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(aes(label = gene_name)) +
#  xlab("Log fold change during acute infection") +
#  ylab("Log fold change during chronic infection") +
  scale_color_manual(values = c("lightgrey", "#FEC287", "lightgrey"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Expression of Toll and Imd pathway genes during acute infection"))
#ggsave(filename = "Results/acute/toll_imd_8_plot.png", toll_imd_8_plot, width= 8, height = 6)



toll_imd_24 <- top_24 %>% 
  mutate(gene_name = rownames(.),
         direction = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                          log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated",
                          TRUE ~ "NS")) %>% 
  filter(gene_name %in% toll_imd_genes$SYMBOL) 

(toll_imd_24_plot <- ggplot(toll_imd_24, aes(x = log2FoldChange, y = -log(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(aes(label = gene_name)) +
#  xlab("Log fold change during acute infection") +
#  ylab("Log fold change during chronic infection") +
  scale_color_manual(values = c("lightgrey", "#F56B5C", "lightgrey"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Expression of Toll and Imd pathway genes 24 hours post injection"))
#ggsave(filename = "Results/mid/toll_imd_24_plot.png", toll_imd_24_plot, width= 8, height = 6)



toll_imd_72 <- top_72 %>% 
  mutate(gene_name = rownames(.),
         direction = case_when(log2FoldChange > 1.5 & pvalue < 0.05 ~ "Upregulated",
                          log2FoldChange < -1.5 & pvalue < 0.05 ~ "Downregulated",
                          TRUE ~ "NS")) %>% 
  filter(gene_name %in% toll_imd_genes$SYMBOL) 

(toll_imd_72_plot <- ggplot(toll_imd_72, aes(x = log2FoldChange, y = -log(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(aes(label = gene_name)) +
#  xlab("Log fold change during acute infection") +
#  ylab("Log fold change during chronic infection") +
  scale_color_manual(values = c("lightgrey", "#A02F7F", "lightgrey"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Expression of Toll and Imd pathway genes during chronic infection"))
#ggsave(filename = "Results/chronic/toll_imd_72_plot.png", toll_imd_72_plot, width= 8, height = 6)




gg_top_time <- countdata_dis
colnames(gg_top_time) <- data_colnames
gg_top_time <- gg_top_time %>% 
  mutate(gene_name = rownames(gg_top_time)) %>% 
  filter(gene_name %in% toll_imd_genes$SYMBOL) %>% 
  dplyr::select(-gene_name) %>% 
  filter(!rowMeans(.) == 0)

library(heatmap3)
library(viridis)
library(gridGraphics)
colors <- colorRampPalette( viridis(9) )(28236)


mat_top_time <- as.matrix(gg_top_time)

mat_top_time <- mat_top_time - rowMeans(mat_top_time)

# heatmap3(mat_top_time, trace = "none", col = colors, labCol = NULL, labRow = rownames(mat_top_time), RowSideLabs = F, ColSideLabs = F, mar = c(5, 12), key.title = "Expression", cexRow = 1, scale = "row")
# grid.echo()
# top_time_heatmap <- grid.grab()
# 
# png(filename = "Results/top_time_heatmap.png", width = 1000, height = 2000, units = "px")
# grid.draw(top_time_heatmap)
# dev.off()
```


## Module Clusterprofiler 95

```{r}
# BiocManager::install("clusterProfiler")
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("pathview")
# BiocManager::install("enrichplot")
library(clusterProfiler)
library(enrichplot)
library(ggplot2)

```

```{r}
organism = "org.Dm.eg.db"
#BiocManager::install(organism, character.only = TRUE)
library(organism, character.only = TRUE)
```


```{r}
GO_enrichment_95 <- vector("list")

modules <- "yellow"

  genes <- module_df_95 %>% 
    filter(colors == "yellow")

original_gene_list <- genes$colors
names(original_gene_list) <- genes$gene_name
gene_list <- na.omit(original_gene_list)
gene_list <- names(gene_list)

gse <- enrichGO(gene_list,
                OrgDb = organism,
                ont = "ALL",
                keyType = "SYMBOL"
                )
GO_enrichment_95_yellow <- gse


```

```{r}

modules <- "yellow"

  genes <- module_df_95 %>% 
    filter(colors == "yellow")

original_gene_list <- genes$colors
names(original_gene_list) <- genes$gene_name

ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

gene <- dedup_ids$ENTREZID

kk <- enrichKEGG(gene = gene,
                 organism = "dme",
                 keyType = "ncbi-geneid",
                 pvalueCutoff = 0.05)

KEGG_enrichment_95_yellow <- kk
(kegg_dotplot <- dotplot(kk, showCategory=10) +
  ggtitle(paste("Enriched KEGG pathways enriched in\nyellow module", sep = " ")) +
    theme(aspect.ratio = 1))
#ggsave(filename = paste("Results/DESeq/yellow_kegg_dotplot.png", sep = ""), kegg_dotplot, width = 6.5, height = 6)


```

### Looking in more depth at specific modules
```{r}
modules <- "yellow"

  module_genes <- module_df_95 %>% 
    filter(colors == modules)
  
module_counts <- countdata_pr %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(gene_name %in% module_genes$gene_name) %>% 
  dplyr::select(-gene_name) %>% 
  filter(!rowMeans(.) == 0)


mat_module_counts <- as.matrix(module_counts)

mat_module_counts <- mat_module_counts - rowMeans(mat_module_counts)


gg_module <- as.data.frame(mat_module_counts)%>% 
  mutate(gene_name = rownames(.)) %>% 
  #left_join(mod, by = "gene") %>% 
  pivot_longer(cols = 1:36, names_to = "sample", values_to = "count") %>% 
  separate(sample, into = c("Dissect", "Sex", "Challenge", "Time", "rep")) %>% 
  mutate(Time = as.numeric(Time),
         Challenge = factor(Challenge, levels = c("UC", "Pr")),
         Group = paste(Challenge, Time, sep = "_"),
         Group = factor(Group, levels = c("UC_8", "UC_24", "UC_72", "Pr_8", "Pr_24", "Pr_72")),
         unchal_zero = case_when(Challenge == "UC"~0,
                                 TRUE ~Time),
         sex_gene_time = paste(gene_name, Sex, Time, sep = "_")) %>% 
  group_by(unchal_zero, Sex, gene_name) %>% 
  mutate(avg_count = mean(count))

direction_labs <- c("Negative fold-change", "Positive fold-change")
names(direction_labs) <- c("Negative", "Positive")

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(plot <- ggplot(gg_module, aes(x = Challenge, y = count, color = gene_name, fill = gene_name)) +
  geom_smooth(aes(group = sex_gene_time)) +
  geom_point() +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1) +
  facet_grid(Sex~Time, labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (normalized to average gene count)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of yellow module in carcasses during \nP. rettgeri infection")) +
  theme(aspect.ratio = 1, strip.background = element_rect(fill = "white")))
#ggsave(paste("Results/DESeq/yellow_facet_by_time_expression_plot.png", sep = ""), plot = plot, width = 6.5, height = 4) 

# (plot <- ggplot(gg_module, aes(x = Time, y = count, color = gene_name)) +
#   geom_smooth(aes(group = gene_name, fill = gene_name)) +
#   geom_point() +
#   theme_bw() +
#   theme(legend.position = "none", aspect.ratio = 1) +
#   facet_grid(Challenge~Sex, labeller = labeller(Sex = sex_labs)) +
#   ylab("Gene counts (normalized to zero)") +
#   xlab("Hours post injection") +
#   ggtitle(paste("Expression of", i, "pathway in carcasses during \nP. rettgeri infection")) +
#   theme(aspect.ratio = 1, strip.background = element_rect(fill = "white")))
#ggsave(paste("Results/", i,"/expression_plot.png", sep = ""), plot = plot, width = 6, height = 6) 

```

#### Yellow module (upregulated during infection)
```{r}
yellow_genes <- module_df_95 %>% 
    filter(colors == "yellow")

yellow_counts <- countdata_dis %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(gene_name %in% yellow_genes$gene_name)

yellow_dif <- top_time %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(gene_name %in% yellow_counts$gene_name) %>% 
  left_join(yellow_counts)%>% 
  pivot_longer(cols = data_colnames, names_to = "sample", values_to = "count") %>% 
  separate(sample, into = c("dissect", "Sex", "Challenge", "Time", "rep")) %>% 
  mutate(Time = as.numeric(Time),
         Challenge = factor(Challenge, levels = c("UC", "Pr")))

direction_labs <- c("Downregulated at 8 hours vs UC", "Upregulated at 8 hours vs UC")
names(direction_labs) <- c("Down-regulated", "Up-regulated")
direction_24_8_labs <- c("Downregulated at 24 hours vs 8 hours", "Upregulated at 24 hours vs 8 hours")
names(direction_24_8_labs) <- c("Down-regulated", "Up-regulated")

ggplotly(ggplot((yellow_dif), aes(x = Challenge, y = count, fill = gene_name, color = gene_name, label = gene_name)) +
  geom_smooth(aes(group = paste(gene_name, Sex, Time))) +
  geom_point() +
  facet_wrap(Sex~Time))

(yellow_raw <- ggplot((yellow_dif), aes(x = Challenge, y = count, fill = gene_name, color = gene_name, label = gene_name)) +
  geom_smooth(aes(group = paste(gene_name, Sex, Time))) +
  geom_point() +
  facet_wrap(Sex~Time) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1))

yellow_8 <- top_8 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_8_vol <- ggplot(yellow_8, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_8), max.overlaps = 5) +
  scale_color_manual(values = c("lightgrey", "#FEC287"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Genes in yellow module during acute infection (8 hours)") +
  ylim(c(0, 135)))


yellow_24 <- top_24 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_24_vol <- ggplot(yellow_24, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_8), max.overlaps = 1) +
  scale_color_manual(values = c("lightgrey", "#F56B5C"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Genes in yellow module during intermediate infection (24 hours)") +
  ylim(c(0, 135)))

yellow_72 <- top_72 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_72_vol <- ggplot(yellow_72, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_72), max.overlaps = 5) +
  scale_color_manual(values = c("lightgrey", "#A02F7F"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Genes in yellow module during chronic infection (72 hours)") +
  ylim(c(0, 135)))

ggsave("Results/DESeq/yellow_vol_8_plot.png", plot = yellow_8_vol, width = 6, height = 5) 
ggsave("Results/DESeq/yellow_vol_24_plot.png", plot = yellow_24_vol, width = 6, height = 5) 
ggsave("Results/DESeq/yellow_vol_72_plot.png", plot = yellow_72_vol, width = 6, height = 5) 

yellow_F_8 <- top_F_8 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_F_8_vol <- ggplot(yellow_F_8, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_F_8)) +
  scale_color_manual(values = c("lightgrey", "#FEC287"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Genes in yellow module during acute infection in females (8 hours)") +
  ylim(c(0, 185)))


yellow_F_24 <- top_F_24 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_F_24_vol <- ggplot(yellow_F_24, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_F_24)) +
  scale_color_manual(values = c("lightgrey", "#F56B5C"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("log2FoldChange") +
  ggtitle("Genes in yellow module during infection in females (24 hours)") +
  ylim(c(0, 185)))


yellow_F_72 <- top_F_72 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_F_72_vol <- ggplot(yellow_F_72, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_F_72)) +
  scale_color_manual(values = c("lightgrey", "#A02F7F"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("log2FoldChange") +
  ggtitle("Genes in yellow module during chronic infection in females (72 hours)") +
  ylim(c(0, 185)))

ggsave("Results/DESeq/yellow_vol_F_8_plot.png", plot = yellow_F_8_vol, width = 6, height = 5) 
ggsave("Results/DESeq/yellow_vol_F_24_plot.png", plot = yellow_F_24_vol, width = 6, height = 5) 
ggsave("Results/DESeq/yellow_vol_F_72_plot.png", plot = yellow_F_72_vol, width = 6, height = 5) 


yellow_M_8 <- top_M_8 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_M_8_vol <- ggplot(yellow_M_8, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_M_8)) +
  scale_color_manual(values = c("lightgrey", "#FEC287"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("log2FoldChange") +
  ggtitle("Genes in yellow module during acute infection in males (8 hours)") +
  ylim(c(0, 185)))


yellow_M_24 <- top_M_24 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_M_24_vol <- ggplot(yellow_M_24, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_M_24)) +
  scale_color_manual(values = c("lightgrey", "#F56B5C"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("log2FoldChange") +
  ggtitle("Genes in yellow module during infection in males (24 hours)") +
  ylim(c(0, 185)))


yellow_M_72 <- top_M_72 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_M_72_vol <- ggplot(yellow_M_72, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_M_72)) +
  scale_color_manual(values = c("#000004", "lightgrey", "#A02F7F"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab("log2FoldChange") +
  ggtitle("Genes in yellow module during chronic infection in males (72 hours)") +
  ylim(c(0, 185)))

ggsave("Results/DESeq/yellow_vol_M_8_plot.png", plot = yellow_M_8_vol, width = 6, height = 5) 
ggsave("Results/DESeq/yellow_vol_M_24_plot.png", plot = yellow_M_24_vol, width = 6, height = 5) 
ggsave("Results/DESeq/yellow_vol_M_72_plot.png", plot = yellow_M_72_vol, width = 6, height = 5) 


yellow_24 <- top_24 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_24_vol <- ggplot(yellow_24, aes(x = log2FoldChange, y = -log(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_8)) +
  scale_color_manual(values = c("lightgrey", "#F56B5C"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Genes in yellow module during intermediate infection (24 hours)"))

yellow_72 <- top_72 %>% 
  filter(rownames(.) %in% yellow_genes$gene_name) %>% 
  mutate(direction = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                               log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                               TRUE ~ "NS"))
(yellow_72_vol <- ggplot(yellow_72, aes(x = log2FoldChange, y = -log(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(yellow_72)) +
  scale_color_manual(values = c("lightgrey", "#A02F7F"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Genes in yellow module during chronic infection (72 hours)"))
ggsave("Results/yellow/vol_8_plot.png", plot = yellow_8_vol, width = 6, height = 5) 
ggsave("Results/yellow/vol_24_plot.png", plot = yellow_24_vol, width = 6, height = 5) 
ggsave("Results/yellow/vol_72_plot.png", plot = yellow_72_vol, width = 6, height = 5) 

acute_immune_response <- yellow_dif %>% 
  filter(direction_8_UC == "Up-regulated_8_vs_UC" & direction_24_vs_8 == "Down-regulated_24_vs_8" & direction_72_vs_24 == "Down-regulated_72_vs_24")
acute_immune_genes <- unique(acute_immune_response$gene_name)
peak_immune_24 <-  yellow_dif %>% 
  filter(direction_8_UC == "Up-regulated_8_vs_UC" & direction_24_vs_8 == "Up-regulated_24_vs_8" & direction_72_vs_24 == "Down-regulated_72_vs_24")
peak_24_genes <- unique(peak_immune_24$gene_name)

up_down_up_immune <-  yellow_dif %>% 
  filter(direction_8_UC == "Up-regulated_8_vs_UC" & direction_24_vs_8 == "Down-regulated_24_vs_8" & direction_72_vs_24 == "Up-regulated_72_vs_24")
up_down_up_genes <- unique(up_down_up_immune$gene_name)

all_acute_yellow_genes <- c(acute_immune_genes, up_down_up_genes)

chronic_immune <- yellow_dif %>% 
  filter(direction_8_UC == "Up-regulated_8_vs_UC" & direction_24_vs_8 == "Up-regulated_24_vs_8" & direction_72_vs_24 == "Up-regulated_72_vs_24")
chronic_immune_genes <- unique(chronic_immune$gene_name)
```


# Sex-specific responses
## Volcano plot
```{r}
mutate <- dplyr::mutate
inter_volcano_plots <- vector("list")
times <- c("8", "24", "72")
colors_vol <- c("#D3C164", "lightgrey", "#05366E")
for(i in times){
  
library(ggrepel)
top <- top_each_time_inter[[i]]
top$gene_name <- rownames(top)


top_vol <- top %>% 
  mutate(color = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Positive fold-change",
                           log2FoldChange < -2 & pvalue < 0.05 ~ "Negative fold-change",
                           TRUE ~ "NS"))

(time_volcano <- top_vol %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = color)) +
  geom_text_repel(data = top_vol %>% filter(!color == "NS"), aes(label = gene_name), max.overlaps = 5) +
  scale_color_manual(values = colors_vol, name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab(paste("Log2 Fold Change at", i, "hours")) +
  ylab("-log10(P value)") +
  ggtitle(paste("Genes differentially expressed in males\nand females during P. rettgeri infection\n(", i, " hours post injection)", sep = "")))
ggsave(filename = paste("Results/DESeq/", i, "_inter_volcano.png", sep = ""), time_volcano, width = 4.5, height = 4)
}

```

## Acute response
```{r}
acute_female <- top_F_8
colnames(acute_female) <- paste(colnames(acute_female),"F",sep="_")
acute_female$gene_name <- rownames(acute_female)
acute_male <- top_M_8 
colnames(acute_male) <- paste(colnames(acute_male),"M",sep="_")
acute_male$gene_name <- rownames(acute_male)

all_acute <- top_inter_8
colnames(all_acute) <- paste(colnames(all_acute), "inter", sep = "_")
all_acute$gene_name <- rownames(all_acute)
all_acute <- all_acute %>% 
   left_join(acute_female, by='gene_name') %>%
                left_join(., acute_male, by='gene_name') 


gg_pos_inter <- countdata_pr %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(all_acute) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep"), remove = FALSE) %>% 
  filter(Time == "8" & Dissect == "Y" & !gene_name == "Drs") %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Sex, Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_F = case_when(log2FoldChange_F > 0 ~ "Upreg (F)",
                                 log2FoldChange_F < 0 ~ "Downreg (F)",
                                 TRUE ~ "NS (F)"),
         direction_M = case_when(log2FoldChange_M > 0 ~ "Upreg (M)",
                                 log2FoldChange_M < 0 ~ "Downreg (M)",
                                 TRUE ~ "NS (M)"),
         direction = paste(direction_F, direction_M)) %>% 
  filter(!is.na(Time_zero)) 

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_pos_inter, aes(x = Time_zero, y = Count)) +
  geom_smooth(aes(fill = gene_name, color = gene_name)) +
  geom_point(aes(y = Count, color = gene_name)) +
  geom_text_repel(data = gg_pos_inter %>% filter(gene_name %in% c("TotM", "TotA", "TotC")), aes(x = 8, label = gene_name)) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction~Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts") +
  xlab("Hours post injection") +
  ggtitle("Expression of genes dimorphically expressed \nat during acute response to P. rettgeri infection \n(positive fold-change)"))
#ggsave("Results/acute/dimorphic_time_plot_8_posfc.png", plot = time_plot, width = 6, height = 14) 


gg_neg_inter <- countdata_pr %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(all_acute) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "8") %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_F = case_when(log2FoldChange_F > 0 ~ "Upreg (F)",
                                 log2FoldChange_F < 0 ~ "Downreg (F)",
                                 TRUE ~ "NS (F)"),
         direction_M = case_when(log2FoldChange_M > 0 ~ "Upreg (M)",
                                 log2FoldChange_M < 0 ~ "Downreg (M)",
                                 TRUE ~ "NS (M)"),
         direction = paste(direction_F, direction_M)) %>% 
  filter(!is.na(Time_zero)) %>% 
  filter(pvalue_inter < 0.05 & log2FoldChange_inter < 0)

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_neg_inter, aes(x = Time_zero, y = Count)) +
  geom_smooth(aes(fill = gene_name, color = gene_name)) +
  geom_point(aes(y = Count, color = gene_name)) +
  geom_text_repel(aes(x = 8, label = gene_name)) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction~Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection") +
  ggtitle("Expression of genes dimorphically expressed \nat during acute response to P. rettgeri infection \n(negative fold-change)"))
#ggsave("Results/acute/dimorphic_time_plot_8_negfc.png", plot = time_plot, width = 6, height = 14) 

upreg_males <- gg_pos_inter %>% 
  filter(direction_M == "Upreg (M)")

higher_in_males_acute <- unique(upreg_males$gene_name)

acute_inter_vol <- top_inter_8 %>% 
  mutate(gene_name = rownames(.)) %>% 
  mutate(direction = case_when(pvalue < 0.05 & log2FoldChange > 1.5 ~ "Up-regulated in males",
                               pvalue < 0.05 & log2FoldChange < -1.5 ~ "Down-regulated in females",
                               TRUE ~ "NS")) %>% 
  filter(gene_name %in% higher_in_males_acute)

(acute_inter_vol_plot <- ggplot(acute_inter_vol, aes(x = log2FoldChange, y = -log(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(acute_inter_vol)) +
  scale_color_manual(values = c("lightgrey", "#05366E", "lightgrey"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Male-specific up-regulation of genes during acute infection"))
#ggsave("Results/acute/male_upreg_vol_plot.png", plot = acute_inter_vol_plot, width = 6, height = 5) 

```


```{r}
original_gene_list <- top_inter_8$log2FoldChange
names(original_gene_list) <- rownames(top_inter_8)
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList = gene_list,
             ont = "ALL",
             keyType = "SYMBOL",
             minGSSize = 3,
             maxGSSize = 800,
             pvalueCutoff = 0.05,
             verbose = TRUE,
             OrgDb = organism,
             pAdjustMethod = "none")

gse_inter_8 <- gse

direction_labs <- c("Positive fold-change", "Negative fold-change")
names(direction_labs) <- c("activated", "suppressed")

(gse_inter_8_dotplot <- dotplot(gse_inter_8, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Dimorphic GO terms during acute infection"))
#ggsave("Results/limma/gse_inter_8_dotplot.png", plot = gse_inter_8_dotplot, width = 14, height = 6) 


ids<-bitr(rownames(top_inter_8), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

df2 = top_inter_8[rownames(top_inter_8) %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_dimorph_8 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

require(DOSE)

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(kegg_8_dotplot <- dotplot(kegg_dimorph_8, showCategory=10, split=".sign", font.size = 8) +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Dimorphic KEGG pathways during acute infection"))
#ggsave("Results/DESeq/dimorphic_kegg_8_dotplot.png", plot = kegg_8_dotplot, width = 6.5, height = 4) 

```

##### Dimorphic genes
```{r}


acute_jakstat_vol <- top_inter_8 %>% 
  mutate(gene_name = rownames(.)) %>% 
  mutate(direction = case_when(pvalue < 0.05 & log2FoldChange > 1.5 ~ "Up-regulated in males",
                               pvalue < 0.05 & log2FoldChange < -1.5 ~ "Up-regulated in females",
                               TRUE ~ "NS")) %>% 
  filter(padj < 0.05)

(acute_jakstat_vol_plot <- ggplot(acute_jakstat_vol, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(acute_jakstat_vol)) +
  scale_color_manual(values = c("lightgrey", "#D3C164", "#05366E"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Sex-specific expression of JAK-STAT genes during acute infection"))
```

#### Normalized like heatmap
```{r}


gg_top_time <- countdata_pr
gg_top_time <- gg_top_time %>% 
  mutate(gene = rownames(gg_top_time)) %>% 
  filter(gene %in% acute_jakstat_vol$gene_name) %>% 
  dplyr::select(-gene)

mat_top_time <- as.matrix(gg_top_time)

mat_top_time <- mat_top_time - rowMeans(mat_top_time)

gg_time <- as.data.frame(mat_top_time) %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_8) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "8" & gene_name %in% jak_stat_genes) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_8 = case_when(log2FoldChange > 0 ~ "Positive fold-change (8 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (8 hours)",
                                 TRUE ~ "No change"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth() +
  geom_point(aes(y = Count)) +
  geom_text_repel(aes(label = gene_name), color = "black", max.overlaps = 3) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_wrap(~Sex, labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts\n(normalized to average gene count)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of JAK-STAT genes at 8 hours\nin carcasses during P. rettgeri infection")))

#ggsave(paste("Results/DESeq/jakstat_time_plot_8.png", sep = ""), plot = time_plot, width = 5.21, height = 3.6) 
```



##### Looking more at innate immune 
```{r}
top_inter_8$gene_name <- rownames(top_inter_8)
pathway <- kegg_dimorph_8@result %>% 
  filter(Description == "Toll and Imd signaling pathway")

genes <- kegg_dimorph_8@geneSets[[pathway$ID]]
genes<-bitr(genes, fromType = "ENTREZID", toType = "SYMBOL", OrgDb=organism)



gg_time <- countdata_pr %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_8) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "8" & gene_name %in% genes$SYMBOL) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_8 = case_when(log2FoldChange > 0 ~ "Positive fold-change (8 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (8 hours)",
                                 TRUE ~ "No change")) %>% 
  filter(!direction_8 == "No change")

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth() +
  geom_point(aes(y = Count)) +
  geom_text_repel(aes(label = gene_name), color = "black") +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction_8~Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of genes in", "Toll and Imd signalling", "\npathways at 8 hours in carcasses during P. rettgeri infection")))
#ggsave("Results/DESeq/acute_toll_imd_plot.png", plot = time_plot, width = 6.5, height = 6.5) 
```



### Counts for each KEGG
```{r}
dimorphic_8 <- c("Ribosome", "Carbon metabolism", "Oxidative phosphorylation", "Citrate cycle (TCA cycle)", "Biosynthesis of amino acids", "2-Oxocarboxylic acid metabolism", "Glycolysis / Gluconeogenesis")
dimorphic_8_dotplot <- vector("list")
pathway_genes <- vector("list")
for(i in dimorphic_8){

pathway <- kegg_dimorph_8@result %>% 
  filter(Description == i)

genes <- kegg_dimorph_8@geneSets[[pathway$ID]]
pathway_genes[[i]] <- genes
gene_names <- bitr(genes, fromType = "ENTREZID", toType = "SYMBOL", OrgDb=organism)

gg_time <- countdata_dis %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_dis)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_8) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "8" & gene_name %in% gene_names$SYMBOL) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_8 = case_when(log2FoldChange > 0 ~ "Positive fold-change (8 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (8 hours)",
                                 TRUE ~ "No change"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth() +
  geom_point(aes(y = Count)) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction_8~Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of genes in", i, "\npathway at 8 hours in carcasses during P. rettgeri infection")))
dimorphic_8_dotplot[[i]] <- time_plot
ggsave(paste("Results/acute/", i,"_dimorphic_time_plot_8.png", sep = ""), plot = time_plot, width = 6, height = 8) 
}
```

#### Normalized like heatmap
```{r}
normalized_dimorphic_8 <- vector("list")
dimorphic_8 <- c("Ribosome", "Carbon metabolism", "Oxidative phosphorylation", "Citrate cycle (TCA cycle)", "Biosynthesis of amino acids", "2-Oxocarboxylic acid metabolism", "Glycolysis / Gluconeogenesis")

for(i in dimorphic_8){

pathway <- kegg_dimorph_8@result %>% 
  filter(Description == i)

genes <- kegg_dimorph_8@geneSets[[pathway$ID]]
pathway_genes[[i]] <- genes
gene_names <- bitr(genes, fromType = "ENTREZID", toType = "SYMBOL", OrgDb=organism)

gg_top_time <- countdata_dis
gg_top_time <- gg_top_time %>% 
  mutate(gene = rownames(gg_top_time)) %>% 
  filter(gene %in% gene_names$SYMBOL) %>% 
  dplyr::select(-gene)

mat_top_time <- as.matrix(gg_top_time)

mat_top_time <- mat_top_time - rowMeans(mat_top_time)

gg_time <- as.data.frame(mat_top_time) %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_dis)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_8) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "8" & gene_name %in% gene_names$SYMBOL) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_8 = case_when(log2FoldChange > 0 ~ "Positive fold-change (8 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (8 hours)",
                                 TRUE ~ "No change"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth() +
  geom_point(aes(y = Count)) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_wrap(Sex~direction_8, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of genes in", i, "pathway at 8 hours\nin carcasses during P. rettgeri infection")))
normalized_dimorphic_8[[i]] <- time_plot
ggsave(paste("Results/limma/", i,"_norm_dimorphic_time_plot_8.png", sep = ""), plot = time_plot, width = 14, height = 6) 
}
```


## Chronic response
##### Dimorphic genes
```{r}


chronic_dimorph <- top_inter_72 %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(pvalue < 0.05)


gg_top_time <- countdata_pr
gg_top_time <- gg_top_time %>% 
  mutate(gene = rownames(gg_top_time)) %>% 
  filter(gene %in% chronic_dimorph$gene_name) %>% 
  dplyr::select(-gene)

mat_top_time <- as.matrix(gg_top_time)

mat_top_time <- mat_top_time - rowMeans(mat_top_time)

gg_time <- as.data.frame(mat_top_time) %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_72) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "72" & gene_name %in% chronic_dimorph$gene_name) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_72 = case_when(log2FoldChange > 0 ~ "Positive fold-change (72 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (72 hours)",
                                 TRUE ~ "No change"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth() +
  geom_point(aes(y = Count)) +
  geom_text_repel(aes(label = gene_name), color = "black", max.overlaps = 30) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction_72~Sex, labeller = labeller(Sex = sex_labs), scales = "free") +
  ylab("Gene counts (normalized to average gene count)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of dimorphic genes at 72 hours\nin carcasses during P. rettgeri infection")))

#ggsave(paste("Results/DESeq/dimorph_time_plot_72.png", sep = ""), plot = time_plot, width = 5.5, height = 5.5) 
```



```{r}
chronic_female <- top_F_72
colnames(chronic_female) <- paste(colnames(chronic_female),"F",sep="_")
chronic_female$gene_name <- rownames(chronic_female)
chronic_male <- top_M_72 
colnames(chronic_male) <- paste(colnames(chronic_male),"M",sep="_")
chronic_male$gene_name <- rownames(chronic_male)

all_chronic <- top_inter_72
colnames(all_chronic) <- paste(colnames(all_chronic), "inter", sep = "_")
all_chronic$gene_name <- rownames(all_chronic)
all_chronic <- all_chronic %>% 
   left_join(chronic_female, by='gene_name') %>%
                left_join(., chronic_male, by='gene_name') 


gg_pos_inter <- countdata_dis %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_dis)), names_to = "Sample", values_to = "Count") %>% 
  left_join(all_chronic) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep"), remove = FALSE) %>% 
  filter(Time == "72") %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Sex, Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_F = case_when(log2FoldChange_F > 0 ~ "Upreg (F)",
                                 log2FoldChange_F < 0 ~ "Downreg (F)",
                                 TRUE ~ "NS (F)"),
         direction_M = case_when(log2FoldChange_M > 0 ~ "Upreg (M)",
                                 log2FoldChange_M < 0 ~ "Downreg (M)",
                                 TRUE ~ "NS (M)"),
         direction = paste(direction_F, direction_M)) %>% 
  filter(!is.na(Time_zero)) %>% 
  filter(pvalue_inter < 0.05 & log2FoldChange_inter > 0)

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_pos_inter, aes(x = Time_zero, y = Count)) +
  geom_smooth(aes(fill = gene_name, color = gene_name)) +
  geom_point(aes(y = Count, color = gene_name)) +
  geom_text_repel(aes(x = 72, label = gene_name)) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction~Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection") +
  ggtitle("Expression of genes dimorphically expressed \nat during chronic response to P. rettgeri infection \n(positive fold-change)"))
#ggsave("Results/chronic/dimorphic_time_plot_72_posfc.png", plot = time_plot, width = 6, height = 14) 


gg_neg_inter <- countdata_dis %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_dis)), names_to = "Sample", values_to = "Count") %>% 
  left_join(all_chronic) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "72") %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_F = case_when(log2FoldChange_F > 0 ~ "Upreg (F)",
                                 log2FoldChange_F < 0 ~ "Downreg (F)",
                                 TRUE ~ "NS (F)"),
         direction_M = case_when(log2FoldChange_M > 0 ~ "Upreg (M)",
                                 log2FoldChange_M < 0 ~ "Downreg (M)",
                                 TRUE ~ "NS (M)"),
         direction = paste(direction_F, direction_M)) %>% 
  filter(!is.na(Time_zero)) %>% 
  filter(pvalue_inter < 0.05 & log2FoldChange_inter < 0)

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_neg_inter, aes(x = Time_zero, y = Count)) +
  geom_smooth(aes(fill = gene_name, color = gene_name)) +
  geom_point(aes(y = Count, color = gene_name)) +
  geom_text_repel(aes(x = 72, label = gene_name)) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction~Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection") +
  ggtitle("Expression of genes dimorphically expressed \nat during chronic response to P. rettgeri infection \n(negative fold-change)"))
#ggsave("Results/chronic/dimorphic_time_plot_72_negfc.png", plot = time_plot, width = 6, height = 14) 

upreg_males <- gg_pos_inter %>% 
  filter(direction_M == "Upreg (M)")

higher_in_males_chronic <- unique(upreg_males$gene_name)

chronic_inter_vol <- top_inter_72 %>% 
  mutate(gene_name = rownames(.)) %>% 
  mutate(direction = case_when(pvalue < 0.05 & log2FoldChange > 1.5 ~ "Up-regulated in males",
                               pvalue < 0.05 & log2FoldChange < -1.5 ~ "Down-regulated in females",
                               TRUE ~ "NS")) %>% 
  filter(gene_name %in% higher_in_males_chronic)

(chronic_inter_vol_plot <- ggplot(chronic_inter_vol, aes(x = log2FoldChange, y = -log(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(chronic_inter_vol)) +
  scale_color_manual(values = c("lightgrey", "#05366E", "lightgrey"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Male-specific up-regulation of genes during chronic infection"))
#ggsave("Results/chronic/male_upreg_vol_plot.png", plot = chronic_inter_vol_plot, width = 6, height = 5) 

```


### GSEA clusterprofiler
```{r}
original_gene_list <- upreg_males$log2FoldChange_inter
names(original_gene_list) <- upreg_males$gene_name
gene_list <- na.omit(original_gene_list)
gene_list <- names(gene_list)

gse <- enrichGO(gene_list,
                OrgDb = organism,
                ont = "ALL",
                keyType = "SYMBOL"
                )
gse_chronic_male_up <- gse


ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

gene <- dedup_ids$ENTREZID

kk <- enrichKEGG(gene = gene,
                 organism = "dme",
                 keyType = "ncbi-geneid",
                 pvalueCutoff = 0.05)

```







```{r}

chronic_inter_vol <- top_inter_72 %>% 
  mutate(direction = case_when(pvalue < 0.05 & log2FoldChange > 1.5 ~ "Male-specific",
                               pvalue < 0.05 & log2FoldChange < -1.5 ~ "Female-specific",
                               TRUE ~ "NS"))

(chronic_inter_vol_plot <- ggplot(chronic_inter_vol, aes(x = log2FoldChange, y = -log(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(chronic_inter_vol)) +
  scale_color_manual(values = c("#D3C164", "#05366E", "lightgrey"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Dimorphic expression of genes during chronic infection"))
#ggsave("Results/limma/chronic_inter_vol_plot.png", plot = chronic_inter_vol_plot, width = 6, height = 5) 


top_inter_72$gene_name <- rownames(top_inter_72)

gg_time <- countdata_dis %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_dis)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_72) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "72") %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_8 = case_when(log2FoldChange > 0 ~ "Positive fold-change (72 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (72 hours)",
                                 TRUE ~ "No change")) %>% 
  filter(!is.na(Time_zero)) %>% 
  filter(pvalue < 0.05)

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth() +
  geom_point(aes(y = Count)) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_wrap(Sex~direction_8, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection") +
  ggtitle("Expression of genes dimorphically expressed \nat 72 hours in carcasses during P. rettgeri infection"))
#ggsave("Results/limma/dimorphic_time_plot_72.png", plot = time_plot, width = 8, height = 6) 

```


### GSEA clusterprofiler

```{r}
original_gene_list <- top_inter_72$log2FoldChange
names(original_gene_list) <- rownames(top_inter_72)
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList = gene_list,
             ont = "ALL",
             keyType = "SYMBOL",
             minGSSize = 3,
             maxGSSize = 800,
             pvalueCutoff = 0.05,
             verbose = TRUE,
             OrgDb = organism,
             pAdjustMethod = "none")

gse_inter_72 <- gse

direction_labs <- c("Positive fold-change", "Negative fold-change")
names(direction_labs) <- c("activated", "suppressed")

(gse_inter_72_dotplot <- dotplot(gse_inter_72, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Dimorphic GO terms during chronic infection"))
#ggsave("Results/limma/gse_inter_8_dotplot.png", plot = gse_inter_8_dotplot, width = 14, height = 6) 



ids<-bitr(rownames(top_inter_72), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

df2 = top_inter_72[rownames(top_inter_72) %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_dimorph_72 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               keyType       = "ncbi-geneid")

require(DOSE)

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(kegg_72_dotplot <- dotplot(kegg_dimorph_72, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Dimorphic gene sets in\nDrosophila 72 hours\npost P. rettgeri injection"))
#ggsave("Results/DESeq/dimorphic_kegg_72_dotplot.png", plot = kegg_72_dotplot, width = 5.21, height = 3.21) 
```


#### Counts for each KEGG
```{r}
dimorphic_72 <- unique(kegg_dimorph_72@result$Description)
dimorphic_72_dotplot <- vector("list")
pathway_genes <- vector("list")
for(i in dimorphic_72){

pathway <- kegg_dimorph_72@result %>% 
  filter(Description == i)

genes <- kegg_dimorph_72@geneSets[[pathway$ID]]
pathway_genes[[i]] <- genes
gene_names <- bitr(genes, fromType = "ENTREZID", toType = "SYMBOL", OrgDb=organism)

gg_time <- countdata_dis %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_dis)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_72) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "72" & gene_name %in% gene_names$SYMBOL) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_72 = case_when(log2FoldChange > 0 ~ "Positive fold-change (72 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (72 hours)",
                                 TRUE ~ "No change"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth() +
  geom_point(aes(y = Count)) +
  geom_text_repel(aes(label = gene_name), color = "black") +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction_72~Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of genes in", i, "\npathway at 72 hours in carcasses during P. rettgeri infection")))
dimorphic_72_dotplot[[i]] <- time_plot
ggsave(paste("Results/chronic/", i,"_dimorphic_time_plot_72.png", sep = ""), plot = time_plot, width = 6, height = 8) 
}
```

#### Normalized like heatmap
```{r}
normalized_dimorphic_72 <- vector("list")

for(i in dimorphic_72){

pathway <- kegg_dimorph_72@result %>% 
  filter(Description == i)

genes <- kegg_dimorph_72@geneSets[[pathway$ID]]
pathway_genes[[i]] <- genes
gene_names <- bitr(genes, fromType = "ENTREZID", toType = "SYMBOL", OrgDb=organism)

gg_top_time <- countdata_dis
gg_top_time <- gg_top_time %>% 
  mutate(gene = rownames(gg_top_time)) %>% 
  filter(gene %in% gene_names$SYMBOL) %>% 
  dplyr::select(-gene)

mat_top_time <- as.matrix(gg_top_time)

mat_top_time <- mat_top_time - rowMeans(mat_top_time)

gg_time <- as.data.frame(mat_top_time) %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_dis)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_72) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "72" & gene_name %in% gene_names$SYMBOL) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_72 = case_when(log2FoldChange > 0 ~ "Positive fold-change (72 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (72 hours)",
                                 TRUE ~ "No change"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth() +
  geom_point(aes(y = Count)) +
  geom_text_repel(aes(label = gene_name), color = "black") +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction_72 ~ Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (normalized to average gene count)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of genes in", i, "\nat 72 hours in carcasses during P. rettgeri infection")))
normalized_dimorphic_72[[i]] <- time_plot
#ggsave(paste("Results/chronic/", i,"_norm_dimorphic_time_plot_72.png", sep = ""), plot = time_plot, width = 6, height = 8) 
}
```

##### Toll and Imd volcano
```{r}
dimorphic_72 <- "Toll and Imd signaling pathway"
pathway <- kegg_dimorph_8@result %>% 
  filter(Description == dimorphic_72)

genes <- kegg_dimorph_8@geneSets[[pathway$ID]]
gene_names <- bitr(genes, fromType = "ENTREZID", toType = "SYMBOL", OrgDb=organism)

gg_top_time <- countdata_pr
gg_top_time <- gg_top_time %>% 
  mutate(gene = rownames(gg_top_time)) %>% 
  filter(gene %in% gene_names$SYMBOL) %>% 
  dplyr::select(-gene)

mat_top_time <- as.matrix(gg_top_time)

mat_top_time <- mat_top_time - rowMeans(mat_top_time)

gg_time <- as.data.frame(mat_top_time) %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_72) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "72" & gene_name %in% gene_names$SYMBOL) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_72 = case_when(log2FoldChange > 0 ~ "Positive fold-change (72 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (72 hours)",
                                 TRUE ~ "No change"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth() +
  geom_point(aes(y = Count)) +
  geom_text_repel(aes(label = gene_name), color = "black") +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(~ Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (normalized to zero)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of genes in Toll and Imd pathways\nat 72 hours in carcasses during P. rettgeri infection")))

#ggsave(paste("Results/DESeq/toll_imd_norm_dimorphic_time_plot_72.png", sep = ""), plot = time_plot, width = 6.5, height = 4) 

```
