---
title: "acute_vs_chronic"
author: "Becca Belmonte"
date: "2022-12-27"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Libraries
```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(DESeq2)
library(ggrepel)
library(WGCNA)
library(viridis)
library(clusterProfiler)
library(enrichplot)
library(ggplot2)
library(plotly)
organism = "org.Dm.eg.db"
#BiocManager::install(organism, character.only = TRUE)
library(organism, character.only = TRUE)
select <- dplyr::select
```


```{r Colors, number=TRUE}
#viridis(10,option="magma")
colors_time <- c("#A02F7F","#F56B5C", "#FEC287", "#000004")
colors_sex <- c("#D3C164", "#05366E")
colors_inf <- c("#000004", "#FEC98D")
colors_wound <- c("#F1605D", "#FEC98D")
colors_dis <- c("#451077", "#9F2F7F")
colors_hem <- viridis(4, option = "cividis")

text_sex <- c("Female", "Male")
text_inf <- c("Infected", "Uninfected")
text_wound <- c("Wounded", "Unchallenged")
text_dis <- c("Dissected", "Not Dissected")
text_hem <- c("Internalization", "Maturation", "Migration", "Receptors")
```


```{r Raw Data, number=TRUE}
countdata <- read.csv("Data/david_counts_wide_unambiguous.csv", row.names = 1, sep = ";")

countdata[is.na(countdata)] <- 0

remove <- c("XF:Z:alignment_not_unique", "XF:Z:no_feature")
rrna <- rownames(countdata)[grep("rRNA", rownames(countdata))]

countdata <- countdata[!rownames(countdata) %in% remove,]
countdata <- countdata[!rownames(countdata) %in% rrna,]

t_countdata <- as.data.frame(t(countdata))
t_countdata$ID <- rownames(t_countdata)

treatmentdata <- read.csv("Data/RNAseq_SexDim_ID.csv", sep=";", row.names = 1)
treatmentdata$ID <- rownames(treatmentdata)
not_dissected <- c("RG.Z.sample_165", "RG.Z.sample_61", "RG.Z.sample_166")
dissected <- c("RG.Z.sample_147")

# based on spermatogenesis gene expression (fzo & Dpy-30L2), sample 165 is unambiguously not dissected -- I correct.
treatmentdata$Dissected_Y.N = ifelse(treatmentdata$ID %in% not_dissected,"N",as.character(treatmentdata$Dissected_Y.N))
treatmentdata$Dissected_Y.N = ifelse(treatmentdata$ID %in% dissected,"Y",as.character(treatmentdata$Dissected_Y.N))

#nrow(subset(treatmentdata, Dissected_Y.N=="N"))

#treatmentdata %>% 
#  group_by(Dissected_Y.N) %>% 
#  summarise(rows = n()) 
#treatmentdata %>% 
#  group_by(Treatment) %>% 
#  summarise(rows = n())
#treatmentdata %>% 
#  group_by(Sex) %>% 
#  summarise(rows = n())
#treatmentdata %>% 
#  group_by(Rep) %>% 
 # summarise(rows = n())

RNAseq_data <- left_join(treatmentdata,t_countdata)

RNAseq_data$totalcounts <- rowSums(RNAseq_data[,-(1:8)])

RNAseq_TPM_col <- RNAseq_data[,(1:8)]

RNAseq_TPM_data <- RNAseq_data[,9:16159]/RNAseq_data$totalcounts*10^6
#101.92322
RNAseq_TPM_data <- cbind(RNAseq_TPM_col, RNAseq_TPM_data)

matrix_countdata <- as.matrix(countdata)
colData <- treatmentdata %>% 
  arrange(ID)
rownames(colData) <- colData$ID
#colnames(matrix_countdata) %in% rownames(colData)
colData <- colData %>% 
  dplyr::select(Rep:Dissected_Y.N, ID)

colData <- colData %>% 
  mutate(Group = str_c(Dissected_Y.N, Sex, Treatment, time, Rep, sep = "_"))
rownames(colData) <- colData$ID
colnames(matrix_countdata) <- colData$Group
rownames(colData) <- colData$Group
colData$Rep <- as.factor(colData$Rep)
#write.csv(colData, file = "colData.csv")

countdata <- countdata %>% 
  select(colData$ID)
colnames(countdata) <- colData$Group
rownames(countdata) <- sub("XF:Z:", "", rownames(countdata))


Gene_ID_Flybase <- read.table("Data/fbgn_annotation_ID_fb_2020_01.tsv",header=F,sep="\t")
colnames(Gene_ID_Flybase) <- c("Gene_name","Species","FBgn_ID","Former_FBgn_ID","CG_ID","Former_CG_ID")
Gene_ID_Flybase <- Gene_ID_Flybase %>% 
  filter(Species == "Dmel")
Gene_ID_Flybase$Species <- as.factor(Gene_ID_Flybase$Species)

Drosophila_KEGG <- read.delim("Data/Drosophila_KEGG.gmt")

t_Drosophila_KEGG <- as.data.frame(t(Drosophila_KEGG))
colnames(t_Drosophila_KEGG) <- t_Drosophila_KEGG[1,]
t_Drosophila_KEGG <- as.list(t_Drosophila_KEGG[-c(1,2), ])


Drosophila_REACTOME <- read.delim("Data/Drosophila_REACTOME.gmt")

t_Drosophila_REACTOME <- as.data.frame(t(Drosophila_REACTOME))
colnames(t_Drosophila_REACTOME) <- t_Drosophila_REACTOME[1,]
t_Drosophila_REACTOME <- as.list(t_Drosophila_REACTOME[-c(1,2), ])
```

# DE models
```{r}
############################################################
# Setting up the dissected 0, 8, 24, 72 model

colData$Dissected_Y.N <- as.factor(colData$Dissected_Y.N)
colData$Dissected_Y.N <- relevel(factor(colData$Dissected_Y.N), "Y")
colData_pr <- colData %>% 
  filter(!Treatment == "WD" & Dissected_Y.N == "Y") %>% 
  mutate(Treatment = factor(Treatment, levels = c("UC", "Pr")),
         Sex = factor(Sex, levels = c("F", "M")))

times <- c("8", "24", "72")
top_each_time <- vector("list")
for(i in times){
  colData_loop <- colData_pr %>% 
  filter(time == i)
countdata_loop <- countdata %>% 
  select(colData_loop$Group)
matrix_countdata_loop <- as.matrix(countdata_loop)

dds_loop <- DESeqDataSetFromMatrix(countData = matrix_countdata_loop,
colData = colData_loop,
design = ~  Sex + Treatment)



dds_loop <- estimateSizeFactors(dds_loop)

dds_loop <- DESeq(dds_loop, test="LRT", reduced = ~ Sex)
resTC_loop <- DESeq2::results(dds_loop)

tab_resTC_loop <- as.data.frame(resTC_loop)
tab_resTC_loop$gene_name <- rownames(tab_resTC_loop) 
tab_resTC_loop$gene_name <- sub("XF:Z:","",tab_resTC_loop$gene_name)


tab_resTC_loop <- cbind(tab_resTC_loop,Gene_ID_Flybase[match(tab_resTC_loop$gene_name,Gene_ID_Flybase$Gene_name),"FBgn_ID"])
colnames(tab_resTC_loop)[ncol(tab_resTC_loop)] ="FBgn_ID"

top_each_time[[i]] <- tab_resTC_loop
}
top_8 <- top_each_time[["8"]]
top_24 <- top_each_time[["24"]]
top_72 <- top_each_time[["72"]]




###########################################
# Sex specific responses over time

times <- c("8", "24", "72")
top_each_time_inter <- vector("list")
for(i in times){
  colData_loop <- colData_pr %>% 
  filter(time == i)
countdata_loop <- countdata %>% 
  select(colData_loop$Group)
matrix_countdata_loop <- as.matrix(countdata_loop)

dds_loop <- DESeqDataSetFromMatrix(countData = matrix_countdata_loop,
colData = colData_loop,
design = ~  Sex*Treatment)



dds_loop <- estimateSizeFactors(dds_loop)

dds_loop <- DESeq(dds_loop, test="LRT", reduced = ~ Sex + Treatment)
resTC_loop <- DESeq2::results(dds_loop)

tab_resTC_loop <- as.data.frame(resTC_loop)
tab_resTC_loop$gene_name <- rownames(tab_resTC_loop) 
tab_resTC_loop$gene_name <- sub("XF:Z:","",tab_resTC_loop$gene_name)


tab_resTC_loop <- cbind(tab_resTC_loop,Gene_ID_Flybase[match(tab_resTC_loop$gene_name,Gene_ID_Flybase$Gene_name),"FBgn_ID"])
colnames(tab_resTC_loop)[ncol(tab_resTC_loop)] ="FBgn_ID"

top_each_time_inter[[i]] <- tab_resTC_loop
}
top_inter_8 <- top_each_time_inter[["8"]]
top_inter_24 <- top_each_time_inter[["24"]]
top_inter_72 <- top_each_time_inter[["72"]]


# Loop response at each time point (females)
top_each_time_F <- vector("list")

times <- c("8", "24", "72")
for(i in times){
  colData_loop <- colData_pr %>% 
  filter(time == i & Sex == "F")
countdata_loop <- countdata %>% 
  select(colData_loop$Group)
matrix_countdata_loop <- as.matrix(countdata_loop)

dds_loop <- DESeqDataSetFromMatrix(countData = matrix_countdata_loop,
colData = colData_loop,
design = ~  Treatment)



dds_loop <- estimateSizeFactors(dds_loop)

dds_loop <- DESeq(dds_loop, test="LRT", reduced = ~ 1)
resTC_loop <- DESeq2::results(dds_loop)

tab_resTC_loop <- as.data.frame(resTC_loop)
tab_resTC_loop$gene_name <- rownames(tab_resTC_loop) 
tab_resTC_loop$gene_name <- sub("XF:Z:","",tab_resTC_loop$gene_name)


tab_resTC_loop <- cbind(tab_resTC_loop,Gene_ID_Flybase[match(tab_resTC_loop$gene_name,Gene_ID_Flybase$Gene_name),"FBgn_ID"])
colnames(tab_resTC_loop)[ncol(tab_resTC_loop)] ="FBgn_ID"

top_each_time_F[[i]] <- tab_resTC_loop
}

top_F_8 <- top_each_time_F[["8"]]
top_F_24 <- top_each_time_F[["24"]]
top_F_72 <- top_each_time_F[["72"]]

#####################################################
# Loop response at each time point (males)
top_each_time_M <- vector("list")

times <- c("8", "24", "72")
for(i in times){
  colData_loop <- colData_pr %>% 
  filter(time == i & Sex == "M")
countdata_loop <- countdata %>% 
  select(colData_loop$Group)
matrix_countdata_loop <- as.matrix(countdata_loop)

dds_loop <- DESeqDataSetFromMatrix(countData = matrix_countdata_loop,
colData = colData_loop,
design = ~  Treatment)



dds_loop <- estimateSizeFactors(dds_loop)

dds_loop <- DESeq(dds_loop, test="LRT", reduced = ~ 1)
resTC_loop <- DESeq2::results(dds_loop)

tab_resTC_loop <- as.data.frame(resTC_loop)
tab_resTC_loop$gene_name <- rownames(tab_resTC_loop) 
tab_resTC_loop$gene_name <- sub("XF:Z:","",tab_resTC_loop$gene_name)


tab_resTC_loop <- cbind(tab_resTC_loop,Gene_ID_Flybase[match(tab_resTC_loop$gene_name,Gene_ID_Flybase$Gene_name),"FBgn_ID"])
colnames(tab_resTC_loop)[ncol(tab_resTC_loop)] ="FBgn_ID"

top_each_time_M[[i]] <- tab_resTC_loop
}

top_M_8 <- top_each_time_M[["8"]]
top_M_24 <- top_each_time_M[["24"]]
top_M_72 <- top_each_time_M[["72"]]


```


## WGCNA 95%
```{r}
# Uncomment and modify the following to install any missing packages
# install.packages(c("tidyverse", "magrittr", "WGCNA"))
library(tidyverse)     # tidyverse will pull in ggplot2, readr, other useful libraries
library(magrittr)      # provides the %>% operator
library(WGCNA)   
```

```{r}
library(DESeq2)

colData_pr <- colData_pr %>% 
  mutate(time_factor = factor(time, levels = c("8", "24", "72")),
         Treatment = factor(Treatment, levels = c("UC", "Pr")),
         condition = factor(paste(Sex, Treatment, time_factor, sep = "_"), levels = c("F_UC_8", "M_UC_8", "F_UC_24", "M_UC_24", "F_UC_72", "M_UC_72", "F_Pr_8", "M_Pr_8", "F_Pr_24", "M_Pr_24", "F_Pr_72", "M_Pr_72")))

countdata_pr <- countdata %>% 
  select(colData_pr$Group)
matrix_countdata_pr <- as.matrix(countdata_pr)

dds <- DESeqDataSetFromMatrix(matrix_countdata_pr,
                              colData_pr,
                              design = ~condition)
#> converting counts to integer mode
#> Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
#> design formula are characters, converting to factors

dds <- DESeq(dds)

#> estimating size factors
#> estimating dispersions
#> gene-wise dispersion estimates
#> mean-dispersion relationship
#> final dispersion estimates
#> fitting model and testing
vsd <- varianceStabilizingTransformation(dds)
library(genefilter)      # <= why is this here?
#>
#> Attaching package: 'genefilter'
#> The following objects are masked from 'package:matrixStats':
#>
#>     rowSds, rowVars
#> The following object is masked from 'package:readr':
#>
#>     spec
wpn_vsd <- getVarianceStabilizedData(dds)
rv_wpn <- rowVars(wpn_vsd)
summary(rv_wpn)
#>     Min.  1st Qu.   Median     Mean  3rd Qu.     Max.
#>  0.00000  0.00000  0.00000  0.08044  0.03322 11.14529

q75_wpn <- quantile( rowVars(wpn_vsd), .75)  # <= original
q95_wpn <- quantile( rowVars(wpn_vsd), .95)  # <= changed to 95 quantile to reduce dataset
expr_normalized <- wpn_vsd[ rv_wpn > q95_wpn, ]

expr_normalized[1:5,1:10]
#>                       B-3      B-4      B-5      L-3      L-4      L-5      S-3
#> AC149818.2_FG001 7.600901 7.077399 7.803434 7.220840 7.410408 8.028223 7.160846
#> AC149829.2_FG003 8.782014 8.179876 7.900062 8.299778 7.529891 8.631731 8.055118
#> AC182617.3_FG001 8.047244 7.120668 6.885533 7.501391 7.279413 7.809565 7.184253
#> AC186512.3_FG001 6.901539 7.389644 6.975945 6.859593 7.370816 6.633722 7.798843
#> AC186512.3_FG007 7.919688 7.754506 7.670946 7.417760 7.988427 7.904850 7.484542
#>                       S-4      S-5   B_L1.1
#> AC149818.2_FG001 7.401382 7.345322 6.524435
#> AC149829.2_FG003 8.744502 8.142909 8.240407
#> AC182617.3_FG001 8.140134 6.972400 7.777347
#> AC186512.3_FG001 6.949501 6.952659 6.059033
#> AC186512.3_FG007 8.375664 7.762799 6.335663
dim(expr_normalized)
#> [1] 5486   24

expr_normalized_df <- data.frame(expr_normalized) %>%
  mutate(
    Gene_id = row.names(expr_normalized)
  ) %>%
  pivot_longer(-Gene_id)

# expr_normalized_df %>% ggplot(., aes(x = name, y = value)) +
#   geom_violin() +
#   geom_point() +
#   theme_bw() +
#   theme(
#     axis.text.x = element_text( angle = 90)
#   ) +
#   ylim(0, NA) +
#   labs(
#     title = "Normalized and 75 quantile Expression",
#     x = "treatment",
#     y = "normalized expression"
#   )
```


```{r}
input_mat = t(expr_normalized)

input_mat[1:5,1:10]           # Look at first 5 rows and 10 columns

#library(WGCNA)
allowWGCNAThreads()          # allow multi-threading (optional)
#> Allowing multi-threading with up to 4 threads.

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to = 20, by = 2))

# Call the network topology analysis function
sft = pickSoftThreshold(
  input_mat,             # <= Input data
  #blockSize = 30,
  powerVector = powers,
  verbose = 5
  )

par(mfrow = c(1,2));
cex1 = 0.9;

plot(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     xlab = "Soft Threshold (power)",
     ylab = "Scale Free Topology Model Fit, signed R^2",
     main = paste("Scale independence")
)
text(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     labels = powers, cex = cex1, col = "red"
)
abline(h = 0.90, col = "red")
plot(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     xlab = "Soft Threshold (power)",
     ylab = "Mean Connectivity",
     type = "n",
     main = paste("Mean connectivity")
)
text(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     labels = powers,
     cex = cex1, col = "red")
```


```{r}
picked_power = 12
temp_cor <- cor       
cor <- WGCNA::cor         # Force it to use WGCNA cor function (fix a namespace conflict issue)
netwk <- blockwiseModules(input_mat,                # <= input here

                          # == Adjacency Function ==
                          power = picked_power,                # <= power here
                          networkType = "signed",

                          # == Tree and Block Options ==
                          deepSplit = 2,
                          pamRespectsDendro = F,
                          # detectCutHeight = 0.75,
                          minModuleSize = 30,
                          maxBlockSize = 4000,

                          # == Module Adjustments ==
                          reassignThreshold = 0,
                          mergeCutHeight = 0.25,

                          # == TOM == Archive the run results in TOM file (saves time)
                          saveTOMs = T,
                          saveTOMFileBase = "ER",

                          # == Output Options
                          numericLabels = T,
                          verbose = 3)


cor <- temp_cor     # Return cor function to original namespace
```


```{r}
# Convert labels to colors for plotting
mergedColors = labels2colors(netwk$colors)
# Plot the dendrogram and the module colors underneath
plotDendroAndColors(
  netwk$dendrograms[[1]],
  mergedColors[netwk$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05 )
```

```{r}
module_df_95 <- data.frame(
  gene_name = names(netwk$colors),
  colors = labels2colors(netwk$colors)
)

module_df_95[1:5,]
#>            gene_id    colors
#> 1 AC149818.2_FG001      blue
#> 2 AC149829.2_FG003      blue
#> 3 AC182617.3_FG001      blue
#> 4 AC186512.3_FG001 turquoise
#> 5 AC186512.3_FG007 turquoise

# write_delim(module_df,
#            file = "Results/gene_modules.txt",
#            delim = "\t")
```


```{r}
# Get Module Eigengenes per cluster
MEs0 <- moduleEigengenes(input_mat, mergedColors)$eigengenes

# Reorder modules so similar modules are next to each other
MEs0 <- orderMEs(MEs0)
module_order = names(MEs0) %>% gsub("ME","", .)

# Add treatment names
MEs0$treatment = row.names(MEs0)

# tidy & plot data
mME = MEs0 %>%
  pivot_longer(-treatment) %>%
  mutate(
    name = gsub("ME", "", name),
    name = factor(name, levels = module_order)
  ) %>% 
  filter(name %in% c("yellow"))

mME$treatment = factor(mME$treatment, levels = c("Y_F_UC_8_1", "Y_F_UC_8_2", "Y_F_UC_8_3",
                                   "Y_M_UC_8_1", "Y_M_UC_8_2", "Y_M_UC_8_3", 
                                   "Y_F_Pr_8_1", "Y_F_Pr_8_2", "Y_F_Pr_8_3",
                                   "Y_M_Pr_8_1", "Y_M_Pr_8_2", "Y_M_Pr_8_3",
                                   "Y_F_UC_24_1", "Y_F_UC_24_2", "Y_F_UC_24_3",
                                   "Y_M_UC_24_1", "Y_M_UC_24_2", "Y_M_UC_24_3", 
                                   "Y_F_Pr_24_1", "Y_F_Pr_24_2", "Y_F_Pr_24_3",
                                   "Y_M_Pr_24_1", "Y_M_Pr_24_2", "Y_M_Pr_24_3",
                                   "Y_F_UC_72_1", "Y_F_UC_72_2", "Y_F_UC_72_3",
                                   "Y_M_UC_72_1", "Y_M_UC_72_2", "Y_M_UC_72_3", 
                                   "Y_F_Pr_72_1", "Y_F_Pr_72_2", "Y_F_Pr_72_3",
                                   "Y_M_Pr_72_1", "Y_M_Pr_72_2", "Y_M_Pr_72_3"))

mME <- mME %>% 
  separate(treatment, into = c("Dissected", "Sex", "Treatment", "Time", "Rep"), remove = FALSE)


(module_heatmap <- mME %>% ggplot(., aes(x=treatment, y=name, fill=value)) +
  geom_tile() +
  theme_bw() +
  scale_fill_viridis() +
  theme(axis.text.x = element_text(angle=90)) +
  labs(title = "Module-trait Relationships", y = "Modules", fill="corr") +
  annotate("segment", x = 0, xend = 6.5, y = 1.7, yend = 1.7, colour = "#000004", linewidth = 5) +
  annotate("segment", x = 6.5, xend = 12.5, y = 1.7, yend = 1.7, colour = "#FEC287", linewidth = 5) +
  annotate("segment", x = 12.5, xend = 18.5, y = 1.7, yend = 1.7, colour = "#000004", linewidth = 5) +
  annotate("segment", x = 18.5, xend = 24.5, y = 1.7, yend = 1.7, colour = "#F56B5C", linewidth = 5) +
  annotate("segment", x = 24.5, xend = 30.5, y = 1.7, yend = 1.7, colour = "#000004", linewidth = 5) +
  annotate("segment", x = 30.5, xend = 37, y = 1.7, yend = 1.7, colour = "#A02F7F", linewidth = 5) +
  annotate("segment", x = 0, xend = 3.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 3.5, xend = 6.5, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  annotate("segment", x = 6.5, xend = 9.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 9.5, xend = 12.5, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  annotate("segment", x = 12.5, xend = 15.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 15.5, xend = 18.5, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  annotate("segment", x = 18.5, xend = 21.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 21.5, xend = 24.5, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  annotate("segment", x = 24.5, xend = 27.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 27.5, xend = 30.5, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  annotate("segment", x = 30.5, xend = 33.5, y = 1.6, yend = 1.6, colour = "#D3C164", linewidth = 5) +
  annotate("segment", x = 33.5, xend = 37, y = 1.6, yend = 1.6, colour = "#05366E", linewidth = 5) +
  expand_limits(y = 1.75))
#ggsave(filename = "Results/DESeq/module_heatmap.png", module_heatmap, width = 6.5, height = 3)
```


```{r}
# pick out a few modules of interest here
modules_of_interest = "yellow"

# Pull out list of genes in that module
submod = module_df_95 %>%
  subset(colors %in% modules_of_interest)

row.names(module_df_95) = module_df_95$gene_name

# Get normalized expression for those genes
expr_normalized[1:5,1:10]

subexpr = expr_normalized[submod$gene_name,]

submod_df = data.frame(subexpr) %>%
  mutate(
    gene_name = row.names(.)
  ) %>%
  pivot_longer(-gene_name) %>%
  mutate(
    module = module_df_95[gene_name,]$colors,
    )

submod_df$name = factor(submod_df$name, levels = c("Y_F_UC_8_1", "Y_F_UC_8_2", "Y_F_UC_8_3",
                                   "Y_M_UC_8_1", "Y_M_UC_8_2", "Y_M_UC_8_3", 
                                   "Y_F_Pr_8_1", "Y_F_Pr_8_2", "Y_F_Pr_8_3",
                                   "Y_M_Pr_8_1", "Y_M_Pr_8_2", "Y_M_Pr_8_3",
                                   "Y_F_UC_24_1", "Y_F_UC_24_2", "Y_F_UC_24_3",
                                   "Y_M_UC_24_1", "Y_M_UC_24_2", "Y_M_UC_24_3", 
                                   "Y_F_Pr_24_1", "Y_F_Pr_24_2", "Y_F_Pr_24_3",
                                   "Y_M_Pr_24_1", "Y_M_Pr_24_2", "Y_M_Pr_24_3",
                                   "Y_F_UC_72_1", "Y_F_UC_72_2", "Y_F_UC_72_3",
                                   "Y_M_UC_72_1", "Y_M_UC_72_2", "Y_M_UC_72_3", 
                                   "Y_F_Pr_72_1", "Y_F_Pr_72_2", "Y_F_Pr_72_3",
                                   "Y_M_Pr_72_1", "Y_M_Pr_72_2", "Y_M_Pr_72_3"))

(expression_plot <- submod_df %>% ggplot(., aes(x=name, y=value, group=gene_name)) +
  geom_line(aes(color = module),
            alpha = 0.2) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90)
  ) +
  facet_grid(rows = vars(module)) +
  scale_color_manual(values = "#DBA800") +
  labs(x = "treatment",
       y = "normalized expression"))
#ggsave("Results/DESeq/expression_module.png", expression_plot, width = 6.5, height = 3)
```

```{r}
genes_of_interest = module_df_95 %>%
  subset(colors %in% modules_of_interest)

expr_of_interest = expr_normalized[genes_of_interest$gene_name,]
expr_of_interest[1:5,1:5]


# Only recalculate TOM for modules of interest (faster, altho there's some online discussion if this will be slightly off)
TOM = TOMsimilarityFromExpr(t(expr_of_interest),
                            power = picked_power)
#> TOM calculation: adjacency..
#> ..will use 4 parallel threads.
#>  Fraction of slow calculations: 0.000000
#> ..connectivity..
#> ..matrix multiplication (system BLAS)..
#> ..normalization..
#> ..done.

# Add gene names to row and columns
row.names(TOM) = row.names(expr_of_interest)
colnames(TOM) = row.names(expr_of_interest)

edge_list = data.frame(TOM) %>%
  mutate(
    gene1 = row.names(.)
  ) %>%
  pivot_longer(-gene1) %>%
  dplyr::rename(gene2 = name, correlation = value) %>%
  unique() %>%
  subset(!(gene1==gene2)) %>%
  mutate(
    module1 = module_df_95[gene1,]$colors,
    module2 = module_df_95[gene2,]$colors
  )

head(edge_list)

```


# Responses ignoring sex
## Volcano plot
```{r}
mutate <- dplyr::mutate
volcano_plots <- vector("list")
times <- c("8", "24", "72")
colors_vol <- vector("list")
colors_vol[["8"]] <- c("#000004", "lightgrey", "#FEC287")
colors_vol[["24"]] <- c("#000004", "lightgrey", "#F56B5C")
colors_vol[["72"]] <- c("#000004", "lightgrey", "#A02F7F")
for(i in times){
  
library(ggrepel)
top <- top_each_time[[i]]
top$gene_name <- rownames(top)


top_vol <- top %>% 
  mutate(color = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Upregulated",
                           log2FoldChange < -2 & pvalue < 0.05 ~ "Downregulated",
                           TRUE ~ "NS"))
colors <- colors_vol[[i]]

(time_volcano <- top_vol %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = color)) +
  geom_text_repel(data = top_vol %>% filter(!color == "NS"), aes(label = gene_name), max.overlaps = 5) +
  scale_color_manual(values = colors, name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab(paste("Log2 Fold Change at", i, "hours")) +
  ylab("-log10(P value)") +
  ggtitle(paste("Genes differentially expressed\nin carcasses during P. rettgeri infection", "\n(", i, " hours post injection)", sep = "")))
ggsave(filename = paste("Results/DESeq/", i, "_volcano.png", sep = ""), time_volcano, width = 4.5, height = 4)
}

```

## Time ClusterProfiler GSEA

```{r}
ids<-bitr(rownames(top_8), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

df2 = top_8[rownames(top_8) %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_8 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

require(DOSE)

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(kegg_8_dotplot <- dotplot(kegg_8, showCategory=10, split=".sign", font.size = 7) +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched gene sets in Drosophila 8 hours\npost P. rettgeri injection"))
ggsave("Results/DESeq/kegg_8_dotplot.png", plot = kegg_8_dotplot, width = 6.5, height = 4) 


#################################################
# 24 hours
ids<-bitr(top_24$FBgn_ID, fromType = "FLYBASE", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("FLYBASE")]),]

df2 = top_24[top_24$FBgn_ID %in% dedup_ids$FLYBASE,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_24 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")


direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(kegg_24_dotplot <- dotplot(kegg_24, showCategory=10, split=".sign", font.size = 10) +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched gene sets in Drosophila 24\nhours post P. rettgeri injection"))
#ggsave("Results/DESeq/kegg_24_dotplot.png", plot = kegg_24_dotplot, width = 6.5, height = 4) 



#################################################
# 72 hours
ids<-bitr(top_72$FBgn_ID, fromType = "FLYBASE", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("FLYBASE")]),]

df2 = top_72[top_72$FBgn_ID %in% dedup_ids$FLYBASE,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_72 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")


direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(kegg_72_dotplot <- dotplot(kegg_72, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Enriched gene sets in Drosophila\n72 hours post P. rettgeri injection"))
#ggsave("Results/DESeq/kegg_72_dotplot.png", plot = kegg_72_dotplot, width = 6.5, height = 4) 
```


## Module Clusterprofiler 95

```{r}
# BiocManager::install("clusterProfiler")
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("pathview")
# BiocManager::install("enrichplot")
library(clusterProfiler)
library(enrichplot)
library(ggplot2)

```

```{r}
organism = "org.Dm.eg.db"
#BiocManager::install(organism, character.only = TRUE)
library(organism, character.only = TRUE)
```

```{r}

modules <- "yellow"

  genes <- module_df_95 %>% 
    filter(colors == "yellow")

original_gene_list <- genes$colors
names(original_gene_list) <- genes$gene_name

ids<-bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

gene <- dedup_ids$ENTREZID

kk <- enrichKEGG(gene = gene,
                 organism = "dme",
                 keyType = "ncbi-geneid",
                 pvalueCutoff = 0.05)

KEGG_enrichment_95_yellow <- kk
(kegg_dotplot <- dotplot(kk, showCategory=10) +
  ggtitle(paste("Enriched KEGG pathways enriched in\nyellow module", sep = " ")) +
    theme(aspect.ratio = 1))
#ggsave(filename = paste("Results/DESeq/yellow_kegg_dotplot.png", sep = ""), kegg_dotplot, width = 6.5, height = 6)


```

### Looking in more depth at specific modules
```{r}
modules <- "yellow"

  module_genes <- module_df_95 %>% 
    filter(colors == modules)
  
module_counts <- countdata_pr %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(gene_name %in% module_genes$gene_name) %>% 
  dplyr::select(-gene_name) %>% 
  filter(!rowMeans(.) == 0)


mat_module_counts <- as.matrix(module_counts)

mat_module_counts <- mat_module_counts - rowMeans(mat_module_counts)


gg_module <- as.data.frame(mat_module_counts)%>% 
  mutate(gene_name = rownames(.)) %>% 
  #left_join(mod, by = "gene") %>% 
  pivot_longer(cols = 1:36, names_to = "sample", values_to = "count") %>% 
  separate(sample, into = c("Dissect", "Sex", "Challenge", "Time", "rep")) %>% 
  mutate(Time = as.numeric(Time),
         Challenge = factor(Challenge, levels = c("UC", "Pr")),
         Group = paste(Challenge, Time, sep = "_"),
         Group = factor(Group, levels = c("UC_8", "UC_24", "UC_72", "Pr_8", "Pr_24", "Pr_72")),
         unchal_zero = case_when(Challenge == "UC"~0,
                                 TRUE ~Time),
         sex_gene_time = paste(gene_name, Sex, Time, sep = "_")) %>% 
  group_by(unchal_zero, Sex, gene_name) %>% 
  mutate(avg_count = mean(count))

direction_labs <- c("Negative fold-change", "Positive fold-change")
names(direction_labs) <- c("Negative", "Positive")

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(plot <- ggplot(gg_module, aes(x = Challenge, y = count, color = gene_name, fill = gene_name)) +
  geom_smooth(aes(group = sex_gene_time), method = "lm") +
  geom_point() +
  geom_text_repel(aes(label = gene_name), color = "black") +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1) +
  facet_grid(Sex~Time, labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (normalized to average gene count)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of yellow module in carcasses during \nP. rettgeri infection")) +
  theme(aspect.ratio = 1, strip.background = element_rect(fill = "white")))
#ggsave(paste("Results/DESeq/yellow_facet_by_time_expression_plot.png", sep = ""), plot = plot, width = 6.5, height = 4) 

```


# Sex-specific responses
## Volcano plot
```{r}
mutate <- dplyr::mutate
inter_volcano_plots <- vector("list")
times <- c("8", "24", "72")
colors_vol <- c("#D3C164", "lightgrey", "#05366E")
for(i in times){
  
library(ggrepel)
top <- top_each_time_inter[[i]]
top$gene_name <- rownames(top)


top_vol <- top %>% 
  mutate(color = case_when(log2FoldChange > 2 & pvalue < 0.05 ~ "Positive fold-change",
                           log2FoldChange < -2 & pvalue < 0.05 ~ "Negative fold-change",
                           TRUE ~ "NS"))

(time_volcano <- top_vol %>% 
  ggplot(aes(x = log2FoldChange, y = -log10(pvalue), label = gene_name)) +
  geom_point(aes(color = color)) +
  geom_text_repel(data = top_vol %>% filter(!color == "NS"), aes(label = gene_name), max.overlaps = 5) +
  scale_color_manual(values = colors_vol, name = NULL) +
  theme_bw() +
  theme(aspect.ratio = 1) +
  xlab(paste("Log2 Fold Change at", i, "hours")) +
  ylab("-log10(P value)") +
  ggtitle(paste("Genes differentially expressed in males\nand females during P. rettgeri infection\n(", i, " hours post injection)", sep = "")))
ggsave(filename = paste("Results/DESeq/", i, "_inter_volcano.png", sep = ""), time_volcano, width = 4.5, height = 4)
}

```

## Acute response
```{r}
acute_female <- top_F_8
colnames(acute_female) <- paste(colnames(acute_female),"F",sep="_")
acute_female$gene_name <- rownames(acute_female)
acute_male <- top_M_8 
colnames(acute_male) <- paste(colnames(acute_male),"M",sep="_")
acute_male$gene_name <- rownames(acute_male)

all_acute <- top_inter_8
colnames(all_acute) <- paste(colnames(all_acute), "inter", sep = "_")
all_acute$gene_name <- rownames(all_acute)
all_acute <- all_acute %>% 
   left_join(acute_female, by='gene_name') %>%
                left_join(., acute_male, by='gene_name') 


gg_pos_inter <- countdata_pr %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(all_acute) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep"), remove = FALSE) %>% 
  filter(Time == "8" & Dissect == "Y" & !gene_name == "Drs") %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Sex, Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_F = case_when(log2FoldChange_F > 0 ~ "Upreg (F)",
                                 log2FoldChange_F < 0 ~ "Downreg (F)",
                                 TRUE ~ "NS (F)"),
         direction_M = case_when(log2FoldChange_M > 0 ~ "Upreg (M)",
                                 log2FoldChange_M < 0 ~ "Downreg (M)",
                                 TRUE ~ "NS (M)"),
         direction = paste(direction_F, direction_M)) %>% 
  filter(!is.na(Time_zero)) 

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_pos_inter, aes(x = Time_zero, y = Count)) +
  geom_smooth(aes(fill = gene_name, color = gene_name)) +
  geom_point(aes(y = Count, color = gene_name)) +
  geom_text_repel(data = gg_pos_inter %>% filter(gene_name %in% c("TotM", "TotA", "TotC")), aes(x = 8, label = gene_name)) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction~Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts") +
  xlab("Hours post injection") +
  ggtitle("Expression of genes dimorphically expressed \nat during acute response to P. rettgeri infection \n(positive fold-change)"))
#ggsave("Results/acute/dimorphic_time_plot_8_posfc.png", plot = time_plot, width = 6, height = 14) 


gg_neg_inter <- countdata_pr %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(all_acute) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "8") %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_F = case_when(log2FoldChange_F > 0 ~ "Upreg (F)",
                                 log2FoldChange_F < 0 ~ "Downreg (F)",
                                 TRUE ~ "NS (F)"),
         direction_M = case_when(log2FoldChange_M > 0 ~ "Upreg (M)",
                                 log2FoldChange_M < 0 ~ "Downreg (M)",
                                 TRUE ~ "NS (M)"),
         direction = paste(direction_F, direction_M)) %>% 
  filter(!is.na(Time_zero)) %>% 
  filter(pvalue_inter < 0.05 & log2FoldChange_inter < 0)

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_neg_inter, aes(x = Time_zero, y = Count)) +
  geom_smooth(aes(fill = gene_name, color = gene_name)) +
  geom_point(aes(y = Count, color = gene_name)) +
  geom_text_repel(aes(x = 8, label = gene_name)) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction~Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection") +
  ggtitle("Expression of genes dimorphically expressed \nat during acute response to P. rettgeri infection \n(negative fold-change)"))
#ggsave("Results/acute/dimorphic_time_plot_8_negfc.png", plot = time_plot, width = 6, height = 14) 

upreg_males <- gg_pos_inter %>% 
  filter(direction_M == "Upreg (M)")

higher_in_males_acute <- unique(upreg_males$gene_name)

acute_inter_vol <- top_inter_8 %>% 
  mutate(gene_name = rownames(.)) %>% 
  mutate(direction = case_when(pvalue < 0.05 & log2FoldChange > 1.5 ~ "Up-regulated in males",
                               pvalue < 0.05 & log2FoldChange < -1.5 ~ "Down-regulated in females",
                               TRUE ~ "NS")) %>% 
  filter(gene_name %in% higher_in_males_acute)

(acute_inter_vol_plot <- ggplot(acute_inter_vol, aes(x = log2FoldChange, y = -log(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(acute_inter_vol)) +
  scale_color_manual(values = c("lightgrey", "#05366E", "lightgrey"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Male-specific up-regulation of genes during acute infection"))
#ggsave("Results/acute/male_upreg_vol_plot.png", plot = acute_inter_vol_plot, width = 6, height = 5) 

```


```{r}
original_gene_list <- top_inter_8$log2FoldChange
names(original_gene_list) <- rownames(top_inter_8)
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList = gene_list,
             ont = "ALL",
             keyType = "SYMBOL",
             minGSSize = 3,
             maxGSSize = 800,
             pvalueCutoff = 0.05,
             verbose = TRUE,
             OrgDb = organism,
             pAdjustMethod = "none")

gse_inter_8 <- gse

direction_labs <- c("Positive fold-change", "Negative fold-change")
names(direction_labs) <- c("activated", "suppressed")

(gse_inter_8_dotplot <- dotplot(gse_inter_8, showCategory=10, split=".sign") +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Dimorphic GO terms during acute infection"))
#ggsave("Results/limma/gse_inter_8_dotplot.png", plot = gse_inter_8_dotplot, width = 14, height = 6) 


ids<-bitr(rownames(top_inter_8), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=organism)
 # remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

df2 = top_inter_8[rownames(top_inter_8) %in% dedup_ids$SYMBOL,]
df2$Y = dedup_ids$ENTREZID
kegg_gene_list <- df2$log2FoldChange
names(kegg_gene_list) <- df2$Y
kegg_gene_list<-na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "dme"
kegg_dimorph_8 <- gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

require(DOSE)

direction_labs <- c("Up-regulated", "Down-regulated")
names(direction_labs) <- c("activated", "suppressed")

(kegg_8_dotplot <- dotplot(kegg_dimorph_8, showCategory=10, split=".sign", font.size = 8) +
  facet_grid(.~.sign, labeller = labeller(.sign = direction_labs)) +
  theme(strip.background = element_rect(fill = "white"), aspect.ratio = 1) +
  ggtitle("Dimorphic KEGG pathways during acute infection"))
#ggsave("Results/DESeq/dimorphic_kegg_8_dotplot.png", plot = kegg_8_dotplot, width = 6.5, height = 4) 

```

##### Dimorphic genes
```{r}


acute_jakstat_vol <- top_inter_8 %>% 
  mutate(gene_name = rownames(.)) %>% 
  mutate(direction = case_when(pvalue < 0.05 & log2FoldChange > 1.5 ~ "Up-regulated in males",
                               pvalue < 0.05 & log2FoldChange < -1.5 ~ "Up-regulated in females",
                               TRUE ~ "NS")) %>% 
  filter(padj < 0.05)

(acute_jakstat_vol_plot <- ggplot(acute_jakstat_vol, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(aes(color = direction)) +
  geom_text_repel(label = rownames(acute_jakstat_vol)) +
  scale_color_manual(values = c("lightgrey", "#D3C164", "#05366E"), name = "") +
  theme_bw() +
  theme(aspect.ratio = 1) +
  ggtitle("Sex-specific expression of JAK-STAT genes during acute infection"))
```

#### Normalized like heatmap
```{r}
jak_stat_genes <- c("dome", "hop", "Stat92E", "upd1", "upd2", "upd3", "asrij", "BRWD3", "Cdk2", "Cdk4", "Cnot4", "CycD", "CycE", "Imp", "lin-28", "mask", "Prx4", "Ptp61F", "RanBP3", "RanBP10", "Ken", "NOT4", "Socs36E", "ET", "ESC", "TotM", "TotA", "TotC", "vir-1")

gg_top_time <- countdata_pr
gg_top_time <- gg_top_time %>% 
  mutate(gene = rownames(gg_top_time)) %>% 
  filter(gene %in% acute_jakstat_vol$gene_name) %>% 
  dplyr::select(-gene)

mat_top_time <- as.matrix(gg_top_time)

mat_top_time <- mat_top_time - rowMeans(mat_top_time)

gg_time <- as.data.frame(mat_top_time) %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_8) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "8" & gene_name %in% jak_stat_genes) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_8 = case_when(log2FoldChange > 0 ~ "Positive fold-change (8 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (8 hours)",
                                 TRUE ~ "No change"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth(method = "lm") +
  geom_point(aes(y = Count)) +
  geom_text_repel(aes(label = gene_name), color = "black", max.overlaps = 3) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_wrap(~Sex, labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts\n(normalized to average gene count)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of JAK-STAT genes at 8 hours\nin carcasses during P. rettgeri infection")))

#ggsave(paste("Results/DESeq/jakstat_time_plot_8.png", sep = ""), plot = time_plot, width = 5.21, height = 3.6) 
```



##### Looking more at Toll 
```{r}
top_inter_8$gene_name <- rownames(top_inter_8)
pathway <- kegg_8@result %>% 
  filter(Description == "Toll and Imd signaling pathway")

genes <- kegg_8@geneSets[[pathway$ID]]
genes<-bitr(genes, fromType = "ENTREZID", toType = "SYMBOL", OrgDb=organism)



gg_time <- countdata_pr %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_8) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "8" & gene_name %in% genes$SYMBOL) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_8 = case_when(log2FoldChange > 0 ~ "Positive fold-change (8 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (8 hours)",
                                 TRUE ~ "No change")) %>% 
  filter(!direction_8 == "No change")

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth(method = "lm") +
  geom_point(aes(y = Count)) +
  geom_text_repel(aes(label = gene_name), color = "black") +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction_8~Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (fpkm)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of genes in", "Toll and Imd signalling", "\npathways at 8 hours in carcasses during P. rettgeri infection")))
#ggsave("Results/DESeq/acute_toll_imd_plot.png", plot = time_plot, width = 6.5, height = 6.5) 
```


## Chronic response
##### Dimorphic genes
```{r}


chronic_dimorph <- top_inter_72 %>% 
  mutate(gene_name = rownames(.)) %>% 
  filter(pvalue < 0.05)


gg_top_time <- countdata_pr
gg_top_time <- gg_top_time %>% 
  mutate(gene = rownames(gg_top_time)) %>% 
  filter(gene %in% chronic_dimorph$gene_name) %>% 
  dplyr::select(-gene)

mat_top_time <- as.matrix(gg_top_time)

mat_top_time <- mat_top_time - rowMeans(mat_top_time)

gg_time <- as.data.frame(mat_top_time) %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_72) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "72" & gene_name %in% chronic_dimorph$gene_name) %>% 
  mutate(Time = as.numeric(Time),
         log2FoldChange = -log2FoldChange) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_72 = case_when(log2FoldChange > 0 ~ "Positive fold-change (72 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (72 hours)",
                                 TRUE ~ "No change"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth(method = "lm") +
  geom_point(aes(y = Count)) +
  geom_text_repel(aes(label = gene_name), color = "black", max.overlaps = 30) +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(direction_72~Sex, labeller = labeller(Sex = sex_labs), scales = "free") +
  ylab("Gene counts (normalized to average gene count)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of dimorphic genes at 72 hours\nin carcasses during P. rettgeri infection")))

#ggsave(paste("Results/DESeq/dimorph_time_plot_72.png", sep = ""), plot = time_plot, width = 5.5, height = 5.5) 
```


##### Toll and Imd volcano
```{r}
dimorphic_72 <- "Toll and Imd signaling pathway"
pathway <- kegg_8@result %>% 
  filter(Description == dimorphic_72)

genes <- kegg_8@geneSets[[pathway$ID]]
gene_names <- bitr(genes, fromType = "ENTREZID", toType = "SYMBOL", OrgDb=organism)

gg_top_time <- countdata_pr
gg_top_time <- gg_top_time %>% 
  mutate(gene = rownames(gg_top_time)) %>% 
  filter(gene %in% gene_names$SYMBOL) %>% 
  dplyr::select(-gene)

mat_top_time <- as.matrix(gg_top_time)

mat_top_time <- mat_top_time - rowMeans(mat_top_time)

gg_time <- as.data.frame(mat_top_time) %>% 
  mutate(gene_name = rownames(.)) %>% 
  pivot_longer(all_of(colnames(countdata_pr)), names_to = "Sample", values_to = "Count") %>% 
  left_join(top_inter_72) %>% 
  separate(Sample, into = c("Dissect", "Sex", "Challenge", "Time", "Rep")) %>% 
  filter(Time == "72" & gene_name %in% gene_names$SYMBOL) %>% 
  mutate(Time = as.numeric(Time)) %>% 
  group_by(Challenge, Time, gene_name) %>% 
  mutate(avg_count = mean(Count)) %>% 
  ungroup() %>% 
  mutate(Time_zero = case_when(Challenge == "UC" ~ 0,
                          Challenge == "Pr" ~ Time),
         direction_72 = case_when(log2FoldChange > 0 ~ "Positive fold-change (72 hours)",
                                 log2FoldChange < 0 ~ "Negative fold-change (72 hours)",
                                 TRUE ~ "No change"))

sex_labs <- c("Female", "Male")
names(sex_labs) <- c("F", "M")

(time_plot <- ggplot(gg_time, aes(x = Time_zero, y = Count, color = gene_name, fill = gene_name)) +
  geom_smooth(method = "lm") +
  geom_point(aes(y = Count)) +
  geom_text_repel(aes(label = gene_name), color = "black") +
  theme_bw() +
  theme(legend.position = "none", aspect.ratio = 1, strip.background = element_rect(fill = "white")) +
  facet_grid(~ Sex, scales = "free", labeller = labeller(Sex = sex_labs)) +
  ylab("Gene counts (normalized to zero)") +
  xlab("Hours post injection") +
  ggtitle(paste("Expression of genes in Toll and Imd pathways\nat 72 hours in carcasses during P. rettgeri infection")))

#ggsave(paste("Results/DESeq/toll_imd_norm_dimorphic_time_plot_72.png", sep = ""), plot = time_plot, width = 6.5, height = 4) 

```
